# Video 3

# Dataframes
is a two dimensional data structure, like a 2d array/matrix with rows and columns. 

Lets convert a matrix into a dataframe 

```{r}
vector1 = c(1:12)
matrix1 = matrix(vector1, ncol=4) #create a matrix from the vector 

Students=c("Pragya", "Deepika", "Chandran")
data = data.frame(Students, matrix1) 
data

headers=c("Students", "Geonomics", "Proteomics", "Microbiology", "Biostatistics")
colnames(data)=headers #add column names 
data

# rowwise addition 
## Add a new column with total marks obtained 
data$total_marks=rowSums(data[, c(2, 3, 4, 5)]) #add from second to fifth column 
data

# Add a new column with average marks obtained 
data$average_marks=rowMeans(data[, c(2, 3, 4, 5)])
data
```

# Rloops
is a sequence of instructions that are executed iterative until a condition is met 

## for loops 
```{r}
for (n in 1:10)
{
  print(n*5)
}
```

loop through a vector and print a square of each element

```{r}
num=c(1,2,3,8,10)
for (n in 1:length(num))
{
  print(num[n]^2)
}
```

Breaking of a loop 
```{r}
#Stop the loop if the n is greater than 7
for (n in num)
{
  if (n > 7){
    break
  }
  
  print(n)
  
}
```

Working with files 
<--check minute59 video 4 -- to set a working directory -->
<-- add screenshots when working -->

run the below command to install `readxl` library
```{r}
#install.packages("readxl") #install using rstudio interface(packages)
```

invoke the library into the program 
```{r}
# Import the library
library(readxl)
```

read the files from excel Excel sheet as a dataframe 
```{r}
hg = read_excel("data/Human_genome.xlsx")
```

viewing the data 
```{r}
View(hg)
```

```{r}
class(hg)
```

find the nature of the data 
```{r}
str(hg)
```

checking the existence of null values
```{r}
sum(is.na(hg))
```

# Q1. Calculate the gene density - genes/megabase for each chromosome

converting to megabases 
```{r}
base_Mb = hg$Base_Pairs/1000000
base_Mb
```

rounding off 
```{r}
rbase_Mb=round(base_Mb, digits = 2)
rbase_Mb
```


# Calculate number of genes per megabase per chromosomes 
```{r}
genes_per_megabase = round(hg$Protein_Coding_genes/base_Mb, digits = 2)
genes_per_megabase
```

Adding a new column 
```{r}
hg["genes_per_Mb"]=genes_per_megabase
str(hg)
```

Write the new table to a file locally
```{r}
write.table(hg, 
            file="hg_modified", 
            quote=FALSE, 
            sep='\t', 
            row.names=FALSE, 
            col.names=TRUE)
```
The file will be saved by default on the working directly unless the file path is changed. 

Checking if the new file is saved 
```{r}
file.show("hg_modified")
```

Removing the mitochondrial DNA data 
```{r}
hg_nuc <- subset(hg, Chromosome!="mtDNA")
View(hg_nuc)
```

Sorting data as per a specific column 
ascending order 
```{r}
hg_nuc_ascending = hg_nuc[order(hg_nuc$genes_per_Mb),]
View(hg_nuc_ascending)
```

descending order
```{r}
hg_nuc_descending = hg_nuc[order(-hg_nuc$genes_per_Mb),]
View(hg_nuc_descending)
```

Selecting specific columns of the data
```{r}
sub_hg=hg[,c(1,2)] #1and2 are the row numbers 
sub_hg
View(sub_hg)
```
