# Working with files
## .XLSX, XLS files
### Reading a file to a dataframe 

This chapter explains how to read a file to a data frame using the `readxl` library. The `readxl` package can be installed from the console by `install.packages("readxl")` command.

Lets begin by importing the library
```{r echo=TRUE, include=TRUE}
# Import the library
library(readxl)
```

The `readxl` library can read .xlsx, xls files to a data frame. For this case, Human genome excel file is read. 
```{r}
hg = read_excel("data/Human_genome.xlsx")
```

View the dataset 
```{r}
str(hg)
```

In R, `head()` and `tail()` are functions used to view the first few rows or the last few rows of a data frame respectively. These functions are particularly useful for quickly inspecting the structure and content of a dataset.

Here's how you can use them for this case;
```{r}
head(hg)
```

The `head()` function displays the first few rows of a data frame or a vector. By default, it shows the first 6 rows, but you can specify the number of rows to display using the `n` argument.
```{r}
head(hg, n=10)
```
The tail argument can be used as;
```{r}
tail(hg)
```
to show the last 6 records while specifying `n` value will show the specified number of rows. 

R identifies this type of data as table or data frame. 
```{r}
class(hg)
```

`str()` is used to find the nature of data such as column data types, number of rows and columns, column names , and first records in the dataframe
```{r}
str(hg)
```

`is.na()` is used to check if there is null/blank values in the dataset. If there is it returns TRUE in place of the null point while it return FALSE in place of the value. 
```{r}
sum(is.na(hg))
```
The "hg" data frame has no null values. 

Lets dive into biology to solve some of the questions about genetics 

**Q1. Calculate the gene densityfor each chromosome**

The gene density is calculated in terms of genes per megabase
```{r}
base_Mb = hg$Base_Pairs/1000000
base_Mb
```

To make the values more clear, the results can be rounded off to two decimal place. 
```{r}
rbase_Mb=round(base_Mb, digits = 2)
rbase_Mb
```

**Q2. Calculate number of genes per megabase per chromosomes**

The protein coding genes are divided to the gene density. 
```{r}
genes_per_megabase = round(hg$Protein_Coding_genes/base_Mb, digits = 2)
genes_per_megabase
```

Lets create a new column to the dataframe to the dataset 
```{r}
hg["genes_per_Mb"]=genes_per_megabase
str(hg)
```

### Write a new table to a file locally
The `write.table()` function is used to write table to file on a local machine. The file is saved by deault to the working directory unless the path is specified. 
```{r}
write.table(hg, 
            file="data/hg_modified", 
            quote=FALSE, 
            sep='\t', 
            row.names=FALSE, 
            col.names=TRUE)
```

This file will be saved by default on the working directory unless the file path is changed. 

Checking if the new file is saved. 
```{r}
file.show("data/hg_modified")
```
## .CSV files 
CSV stands for "Comma Separated Values". This is a plain text file that contains data in a tabular format for example the data below that represents athletes' statistics. 
```
Name, Age, Weight
Salman, 30, 75
Pragya, 35, 72
Vidyat, 25, 68
```

Each line in a CSV file represents a row while each field in a row is separated by a comma ",". These files are often used in data manipulations since they are lightweight, simple and widely supported. In this course we shall briefly go through two csv files and do brief analyses. 

### Titanic dataset
The titanic data is a famous data set in data analysis. It contains information about passengers that boarded the ship Titan(not actual true) including their age, ticket class, fare and others. 

#### Reading the csv file

The `read.csv()` function is used to read a cav file to a data frame.
```{r}
# Read data from a CSV file into a data frame
titanic_df = read.csv("data/titanic_dataset.csv")
```

#### Data Assessment

- Structure of the dataset
```{r}
str(titanic_df)
```

- Show the first 6 records of the data set.
```{r}
head(titanic_df)
```

- Count the null values in the data
```{r}
sum(is.na(titanic_df))
```

The are 87 fields in the data that are null(empty)

- Calculate the descriptive statistics
```{r}
summary(titanic_df)
```

#### Data Analysis 
The answers to the two questions will give a brief understanding of titanic passengers.

- **What is the survival rate by gender**
```{r}
# Survival rate by gender
table(titanic_df$Sex, titanic_df$Survived)
```

From the table above, all males drowned but all females survived. 

- **What is the survival rate by ticket class**
```{r}
# Survival rate by ticket class
table(titanic_df$Pclass, titanic_df$Survived)
```

Most passengers from ticket class 0 survived. 






