# Video 6

Chaining 
You can several operations together using the `%>%` or forward pipe operator. <-- add more information-->

Install the dplyr package by `install.packages("dplyr")` on the console, now load all the required libraries 
```{r}
library(readxl)
library(dplyr)
```

Read the dataset from a file into a dataframe
```{r}
hg = read_excel("data/Human_genome.xlsx")
View(hg)
class(hg)
```

review the first six records of the dataset 
```{r}
head(hg)
```
 
 the last six records of the dataset
```{r}
tail(hg)
```
 
Randomly sample fixed number of rows 
```{r}
hg %>% sample_n(5)
```
 
Randomly sample a percent of rows
```{r}
hg %>% sample_frac(0.25, replace = FALSE)
```

#Q1. Which chromosomes has maximum size
- select, arrange, slice 

select the required columns i.e Chromosomes and Base_Pairs(size)
arrange according to Base_Pairs in descending order 
slice to select the first row 

All the above steps should be piped
```{r}
hg1 = hg %>% #transfer the data to a new variable
  select(Chromosome, Base_Pairs) %>% # select the required columns 
           arrange(desc(Base_Pairs)) %>% #order in descending order
           slice(1)

View(hg1)
```

#Q2. List of Chromosomes with more than 500 pseudogenes and more tha 1000 protein coding genes 
-filter 
`&` condition is used in the filter 

```{r}
hg2=hg%>%
  filter(Pseudogenes>500 & Protein_Coding_genes>1000)
View(hg2)
```

#Q3. List of Chromosomes with more tha 500 pseudogenes or more than 1000 protein coding genes and miRNA more than 100
- filter
```{r}
hg2=hg %>% 
  filter((Pseudogenes>500|Protein_Coding_genes>1000) & miRNA>100)
View(hg2)
```

#Q4. Which 10 Chromosomes have the highest protein gene density (Genes per megabase)
- Mutate -> is a command to create a new column 
```{r}
hg3=hg %>%
  mutate(Genes_per_Mb=round((Protein_Coding_genes/(Base_Pairs/100000)),
                            digit=2))
View(hg3)
```

- arrange and select 
```{r}
hg4=hg3 %>%
  arrange(desc(Genes_per_Mb)) %>% # order in descending order 
  select(Chromosome, Genes_per_Mb) %>% # select the required columns 
  slice(1:10) # select the top 10

View(hg4)
```

#Q5. Calculate the mean of gene density for autosomes, sex chromosomes and mitochondrial genes 
```{r}
autosomes=rep("A", 22) #create a vector with 22 As
print(autosomes)
chr_class=c(autosomes, "S", "S", "M")
chr_class
hg5= hg3 %>%
  mutate(Type=chr_class)

View(hg5) # confirm the new column "Type" is added
```

Group by chromosomes and calculate the mean density 
```{r}
hg6=hg5 %>% 
  group_by(Type) %>%
  summarise(gene_density=mean(Genes_per_Mb))
View(hg6)
```

