# Video 8

Check out this link to install `BiocManager` package
https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html

Load the required libraries 
```{r}
library(Biostrings)
```

Read the sequence 
```{r}
seq = readDNAStringSet("data/multifasta_seq.txt")
seq
```

```{r}
seq[[1]]
seq[[2]]
```

Convert to character 
```{r}
as.character(seq[[1]])
```

Number of sequences 
```{r}
length(seq)
```

- Names of the sequences
```{r}
names(seq)
```

first 6 sequence 
```{r}
head(seq)
```

- Reverse compliment of sequence 
```{r}
seq_rc = reverseComplement(seq[[1]])
head(seq_rc)
tail(seq_rc)
as.character(seq_rc)
```

- Sequence composition -default - calculate the DNA composition
```{r}
composition=alphabetFrequency(seq)
print(composition)
```

- Specific composition - lets say we wanna look for A, T, G, C and N
```{r}
composition1=letterFrequency(seq, letters=c("A","T", "G", "C", "N"))
print(composition1)
```

About GC content - here we look for "G' and "C"
```{r}
gc_content <- letterFrequency(seq, letters="GC", as.prob=TRUE)
gc_content
```

Calculate the mean of Gc content 
```{r}
mean(gc_content)
```

Standard deviation 
```{r}
sd(gc_content)
```


```{r}
range(gc_content)
```

Which GC has the maximum content 
```{r}
seq[which(gc_content==max(gc_content))]
```

- Search for substrings 
```{r}
ECoR1=vcountPattern("GAATTC", seq)
print(ECoR1)

HindIII = vcountPattern("AAGTT", seq)
print(HindIII)
```

- Position of substrings
```{r}
HindIII_pos=vmatchPattern("AAGCT", seq)
print(HindIII_pos)
```

- Translate 
```{r}
# translate(seq[[2]])
#Check functions available
methods(class = "DNAStringSet")
```

# Using seqinr

After installing the library by invoking `install.packages("seqinr")` to the console, load it to the workspace 
```{r}
library(seqinr)
```

Reading the fasta file 
```{r}
proseq = read.fasta(file="data/spike_multifasta.txt", seqtype="AA")
View(proseq)
```

Count the number of sequences 
```{r}
num_seq = length(proseq)
print(paste("Number of sequences = ", num_seq))
```

- Names of the sequences
```{r}
names=getName(proseq)
names 
```

- Annotation of sequences 
```{r}
annotation <- getAnnot(proseq)
annotation
```

- Length of individual sequences 
```{r}
len_seq = getLength(proseq)
len_seq
```

- Protein statistics 
```{r}
print(AAstat(proseq[[1]]))
```

It gives the type of proteins 
```{r}
for (i in 1:num_seq)
{
  print(names[i])
  print(AAstat(proseq[[i]]))
}
```

- Show commands in seqinr
```{r}
lseqinr()
```

# Using Short Read
This library can be installed by invoking the `BiocManager::install("ShortRead")` command to the console. 

load the library 
```{r}
library(ShortRead)
```

- Reading the fastq sequences 
```{r}
fq = readFastq("data/file2.fastq")
fq
```

- Preview of first 3 sequences
```{r}
fq[1:3]
```

- Sequence information
```{r}
sread(fq)
```

Subset the sequence from second to fifth
```{r}
sread(fq[2:5])
```

Inspect the quality of the sequence 
```{r}
quality(fq)
```

- Number of sequence in the file 
```{r}
length(fq)
```

- Length of reads 
```{r}
readLength = width(fq)
readLength
```

- generating the summary of fasta sequences 
```{r}
fq_reads = sread(fq)

id_of_reads=id(fq)
id_of_reads[1:5]
fq_reads
```

- Seq length - we can generate the sequence lenght by converting it to a dataframe
```{r}
seq_length = as.data.frame(fq_reads@ranges@width)
seq_length
```

- Quality scores 
```{r}
quality_scores=quality(fq)
quality_scores[1:5]
```

Convert the scores to matrix and see as real numbers 
```{r}
scores=as(quality_scores, "matrix")
scores
```

- Mean score for each sequence 
```{r}
mean_score_for_each_sequence = rowMeans(scores, na.rm=TRUE)
mean_score_for_each_sequence
```

- Mean score for each position 
```{r}
mean_score_for_each_position = rowMeans(scores, na.rm=TRUE)
mean_score_for_each_position
```

Plot a boxplot for the scores 
```{r}
boxplot(scores, xlab="Position", ylab="Phred score")
```

