<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Bioconductor | R Course</title>
  <meta name="description" content="Chapter 7 Bioconductor | R Course" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Bioconductor | R Course" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Bioconductor | R Course" />
  
  
  

<meta name="author" content="YOUR NAME HERE" />


<meta name="date" content="2024-06-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ggplot.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SHORT TITLE HERE</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> R - Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#r---basic-concepts-and-applications"><i class="fa fa-check"></i><b>1.1</b> R - Basic Concepts and applications</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#why-r"><i class="fa fa-check"></i><b>1.2</b> Why R?</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#rstudio---look-and-feel"><i class="fa fa-check"></i><b>1.3.1</b> RStudio - Look and Feel</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#basic-mathematics-like-operators"><i class="fa fa-check"></i><b>1.4</b> Basic Mathematics like operators</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#basics"><i class="fa fa-check"></i><b>1.4.1</b> Basics</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#bodmas-rule"><i class="fa fa-check"></i><b>1.4.2</b> BODMAS rule</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#r-markdown"><i class="fa fa-check"></i><b>1.5</b> R Markdown</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#installation-of-packages"><i class="fa fa-check"></i><b>1.6</b> Installation of packages</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#assignment"><i class="fa fa-check"></i><b>1.7</b> Assignment</a></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#code-challenge"><i class="fa fa-check"></i><b>1.8</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html"><i class="fa fa-check"></i><b>2</b> Data Types and Data Structures</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-types"><i class="fa fa-check"></i><b>2.1</b> Data types</a></li>
<li class="chapter" data-level="2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-structures"><i class="fa fa-check"></i><b>2.2</b> Data Structures</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#vector"><i class="fa fa-check"></i><b>2.2.1</b> Vector</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#matrix"><i class="fa fa-check"></i><b>2.2.2</b> Matrix</a></li>
<li class="chapter" data-level="2.2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-frame"><i class="fa fa-check"></i><b>2.2.3</b> Data frame</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#code-challenge-1"><i class="fa fa-check"></i><b>2.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="loops.html"><a href="loops.html"><i class="fa fa-check"></i><b>3</b> Loops</a>
<ul>
<li class="chapter" data-level="3.1" data-path="loops.html"><a href="loops.html#for-loops"><i class="fa fa-check"></i><b>3.1</b> For loops</a></li>
<li class="chapter" data-level="3.2" data-path="loops.html"><a href="loops.html#while-loop"><i class="fa fa-check"></i><b>3.2</b> While loop</a></li>
<li class="chapter" data-level="3.3" data-path="loops.html"><a href="loops.html#code-challenge-2"><i class="fa fa-check"></i><b>3.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="working-with-files.html"><a href="working-with-files.html"><i class="fa fa-check"></i><b>4</b> Working with files</a>
<ul>
<li class="chapter" data-level="4.1" data-path="working-with-files.html"><a href="working-with-files.html#xlsx-xls-files"><i class="fa fa-check"></i><b>4.1</b> .XLSX, XLS files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="working-with-files.html"><a href="working-with-files.html#reading-a-file-to-a-dataframe"><i class="fa fa-check"></i><b>4.1.1</b> Reading a file to a dataframe</a></li>
<li class="chapter" data-level="4.1.2" data-path="working-with-files.html"><a href="working-with-files.html#write-a-new-table-to-a-file-locally"><i class="fa fa-check"></i><b>4.1.2</b> Write a new table to a file locally</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="working-with-files.html"><a href="working-with-files.html#csv-files"><i class="fa fa-check"></i><b>4.2</b> .CSV files</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="working-with-files.html"><a href="working-with-files.html#titanic-dataset"><i class="fa fa-check"></i><b>4.2.1</b> Titanic dataset</a></li>
<li class="chapter" data-level="4.2.2" data-path="working-with-files.html"><a href="working-with-files.html#ibm-employee-attrition-data"><i class="fa fa-check"></i><b>4.2.2</b> IBM Employee Attrition data</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="working-with-files.html"><a href="working-with-files.html#code-challenge-3"><i class="fa fa-check"></i><b>4.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dpyr-library.html"><a href="dpyr-library.html"><i class="fa fa-check"></i><b>5</b> Dpyr library</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dpyr-library.html"><a href="dpyr-library.html#dplyr-verbs"><i class="fa fa-check"></i><b>5.1</b> Dplyr Verbs</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="dpyr-library.html"><a href="dpyr-library.html#select"><i class="fa fa-check"></i><b>5.1.1</b> select</a></li>
<li class="chapter" data-level="5.1.2" data-path="dpyr-library.html"><a href="dpyr-library.html#filter"><i class="fa fa-check"></i><b>5.1.2</b> filter</a></li>
<li class="chapter" data-level="5.1.3" data-path="dpyr-library.html"><a href="dpyr-library.html#arrange"><i class="fa fa-check"></i><b>5.1.3</b> arrange</a></li>
<li class="chapter" data-level="5.1.4" data-path="dpyr-library.html"><a href="dpyr-library.html#mutate"><i class="fa fa-check"></i><b>5.1.4</b> mutate</a></li>
<li class="chapter" data-level="5.1.5" data-path="dpyr-library.html"><a href="dpyr-library.html#group_by"><i class="fa fa-check"></i><b>5.1.5</b> group_by</a></li>
<li class="chapter" data-level="5.1.6" data-path="dpyr-library.html"><a href="dpyr-library.html#summarise"><i class="fa fa-check"></i><b>5.1.6</b> Summarise</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="dpyr-library.html"><a href="dpyr-library.html#chaining"><i class="fa fa-check"></i><b>5.2</b> Chaining</a></li>
<li class="chapter" data-level="5.3" data-path="dpyr-library.html"><a href="dpyr-library.html#application-of-dplyr-on-microbiology"><i class="fa fa-check"></i><b>5.3</b> Application of Dplyr on Microbiology</a></li>
<li class="chapter" data-level="5.4" data-path="dpyr-library.html"><a href="dpyr-library.html#code-challenge-4"><i class="fa fa-check"></i><b>5.4</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ggplot.html"><a href="ggplot.html"><i class="fa fa-check"></i><b>6</b> GGPLOT</a>
<ul>
<li class="chapter" data-level="6.1" data-path="ggplot.html"><a href="ggplot.html#five-key-steps-to-ggplot"><i class="fa fa-check"></i><b>6.1</b> Five key steps to ggplot</a></li>
<li class="chapter" data-level="6.2" data-path="ggplot.html"><a href="ggplot.html#prepare-the-data-for-plotting---data-assessment"><i class="fa fa-check"></i><b>6.2</b> Prepare the data for plotting - Data Assessment</a></li>
<li class="chapter" data-level="6.3" data-path="ggplot.html"><a href="ggplot.html#plotting"><i class="fa fa-check"></i><b>6.3</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="ggplot.html"><a href="ggplot.html#base-set-up"><i class="fa fa-check"></i><b>6.3.1</b> Base set up</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="ggplot.html"><a href="ggplot.html#working-with-inbuilt-iris-data"><i class="fa fa-check"></i><b>6.4</b> Working with inbuilt iris data</a></li>
<li class="chapter" data-level="6.5" data-path="ggplot.html"><a href="ggplot.html#code-challenge-5"><i class="fa fa-check"></i><b>6.5</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>7</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="7.1" data-path="bioconductor.html"><a href="bioconductor.html#installing-biocmanager-package"><i class="fa fa-check"></i><b>7.1</b> Installing BiocManager Package</a></li>
<li class="chapter" data-level="7.2" data-path="bioconductor.html"><a href="bioconductor.html#searching-and-analyzing-the-strings"><i class="fa fa-check"></i><b>7.2</b> Searching and Analyzing the strings</a></li>
<li class="chapter" data-level="7.3" data-path="bioconductor.html"><a href="bioconductor.html#using-seqinr"><i class="fa fa-check"></i><b>7.3</b> Using seqinr</a></li>
<li class="chapter" data-level="7.4" data-path="bioconductor.html"><a href="bioconductor.html#using-short-read"><i class="fa fa-check"></i><b>7.4</b> Using Short Read</a></li>
<li class="chapter" data-level="7.5" data-path="bioconductor.html"><a href="bioconductor.html#genomic-ranges"><i class="fa fa-check"></i><b>7.5</b> Genomic Ranges</a></li>
<li class="chapter" data-level="7.6" data-path="bioconductor.html"><a href="bioconductor.html#using-deseq-package"><i class="fa fa-check"></i><b>7.6</b> Using `DESeq package</a></li>
<li class="chapter" data-level="7.7" data-path="bioconductor.html"><a href="bioconductor.html#network-analysis-of-genomic-data"><i class="fa fa-check"></i><b>7.7</b> Network Analysis of Genomic Data</a></li>
<li class="chapter" data-level="7.8" data-path="bioconductor.html"><a href="bioconductor.html#code-challenge-6"><i class="fa fa-check"></i><b>7.8</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bioconductor" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Bioconductor<a href="bioconductor.html#bioconductor" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script src="js/dcl.js"></script>
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
<script src="js/dcl.js"></script>
<p>Bioconductor is a repository for high-quality, peer-reviewed software packages for the analysis and comprehension of high-throughput genomic data. It contained specific tools that handle, analyze and visualize data from high-throughput assays for instance ChIP-Seq, single-cell RNA-Seq and RNA-Seq. It provides data structures such as <code>ExpressionSet</code>,<code>SummarizedExperiment</code>, and <code>GenomicRanges</code>that store and manipulate large scale genomic data. It also consists over 2000 packages just for data pre-processing, normalization, data analysis and visualization. Here are some of the popular packages in <code>BioConductor</code> has to offer;</p>
<ul>
<li><p><strong><code>Biostrings</code></strong>: For efficient manipulation of large biological sequences.</p></li>
<li><p><strong><code>DESeq2</code></strong>: For differential gene expression analysis based on count data.</p></li>
<li><p><strong><code>GenomicRanges</code></strong>: For representing and manipulating genomic intervals and variables defined along a genome.</p></li>
<li><p><strong><code>AnnotationDbi</code></strong>: For handling and querying biological annotation databases.</p></li>
<li><p><strong><code>edgeR</code></strong>: Another popular package for differential expression analysis, particularly for RNA-Seq data.</p></li>
</ul>
<div id="installing-biocmanager-package" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Installing BiocManager Package<a href="bioconductor.html#installing-biocmanager-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To get started with Bioconductor, you need to install the <code>BiocManager</code> package, which facilitates the installation and management of Bioconductor packages. You can find detailed installation instructions in the <a href="https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html">official documentation</a>.</p>
Load the required libraries
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KEJpb3N0cmluZ3MpIn0=
</div>
After loading the necessary libraries, you can read sequence data into R. For instance, if you have sequence data stored in a multifasta file named multifasta_seq.txt, you can use the readDNAStringSet() function.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZXEgPSByZWFkRE5BU3RyaW5nU2V0KFwiZGF0YS9tdWx0aWZhc3RhX3NlcS50eHRcIilcbnNlcSJ9
</div>
Each genomic sequence can be viewed separately by indexing
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZXFbWzFdXSAjZmlyc3QgZ2Vub21pYyBzZXF1ZW5jZVxuc2VxW1syXV0gIyBzZWNvbmQgZ2Vub21pYyBzZXF1ZW5jZSJ9
</div>
Convert to character to inspect its content.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJhcy5jaGFyYWN0ZXIoc2VxW1sxXV0pICJ9
</div>
The <code>length</code> function from base R is used to count the number of genomic sequences.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsZW5ndGgoc2VxKSJ9
</div>
<ul>
<li>Names of the sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJuYW1lcyhzZXEpIn0=
</div></li>
<li>view the first 6 sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZWFkKHNlcSkifQ==
</div></li>
<li>Checking the reverse compliment of sequence
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZXFfcmMgPSByZXZlcnNlQ29tcGxlbWVudChzZXFbWzFdXSlcbmhlYWQoc2VxX3JjKVxudGFpbChzZXFfcmMpXG5hcy5jaGFyYWN0ZXIoc2VxX3JjKSJ9
</div></li>
<li>Sequence composition -default - calculate the DNA composition
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJjb21wb3NpdGlvbj1hbHBoYWJldEZyZXF1ZW5jeShzZXEpXG5wcmludChjb21wb3NpdGlvbikifQ==
</div></li>
<li>Specific composition - lets say we wanna look for A, T, G, C and N
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJjb21wb3NpdGlvbjE9bGV0dGVyRnJlcXVlbmN5KHNlcSwgbGV0dGVycz1jKFwiQVwiLFwiVFwiLCBcIkdcIiwgXCJDXCIsIFwiTlwiKSlcbnByaW50KGNvbXBvc2l0aW9uMSkifQ==
</div></li>
</ul>
Computing the GC Content
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJnY19jb250ZW50IDwtIGxldHRlckZyZXF1ZW5jeShzZXEsIGxldHRlcnM9XCJHQ1wiLCBhcy5wcm9iPVRSVUUpXG5nY19jb250ZW50In0=
</div>
Calculate the mean of Gc content
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJtZWFuKGdjX2NvbnRlbnQpIn0=
</div>
<ul>
<li>Standard deviation
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZChnY19jb250ZW50KSJ9
</div></li>
</ul>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJyYW5nZShnY19jb250ZW50KSJ9
</div>
<ul>
<li>Which GC has the maximum content
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZXFbd2hpY2goZ2NfY29udGVudD09bWF4KGdjX2NvbnRlbnQpKV0ifQ==
</div></li>
</ul>
</div>
<div id="searching-and-analyzing-the-strings" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Searching and Analyzing the strings<a href="bioconductor.html#searching-and-analyzing-the-strings" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A biologist can search for specific substrings within the sequences and analyze their positions and characteristics for instance;</p>
<ul>
<li>Search for substrings- in this case we search for <code>"GAATTC"</code> substring.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJFQ29SMT12Y291bnRQYXR0ZXJuKFwiR0FBVFRDXCIsIHNlcSlcbnByaW50KEVDb1IxKVxuXG5IaW5kSUlJID0gdmNvdW50UGF0dGVybihcIkFBR1RUXCIsIHNlcSlcbnByaW50KEhpbmRJSUkpIn0=
</div></li>
<li>Finding the position of substrings
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJIaW5kSUlJX3Bvcz12bWF0Y2hQYXR0ZXJuKFwiQUFHQ1RcIiwgc2VxKVxucHJpbnQoSGluZElJSV9wb3MpIn0=
</div></li>
<li>Translate
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIHRyYW5zbGF0ZShzZXFbWzJdXSlcbiNDaGVjayBmdW5jdGlvbnMgYXZhaWxhYmxlXG5tZXRob2RzKGNsYXNzID0gXCJETkFTdHJpbmdTZXRcIikifQ==
</div></li>
</ul>
</div>
<div id="using-seqinr" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Using seqinr<a href="bioconductor.html#using-seqinr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
Another useful library in sequence analysis is <code>seqinr</code>.
After installing the library by invoking <code>install.packages("seqinr")</code> to the console, load it to the workspace
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KHNlcWlucikifQ==
</div>
Reading the fasta file containing the protein sequence using the <code>read.fasta</code> function.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJwcm9zZXEgPSByZWFkLmZhc3RhKGZpbGU9XCJkYXRhL3NwaWtlX211bHRpZmFzdGEudHh0XCIsIHNlcXR5cGU9XCJBQVwiKVxuc3RyKHByb3NlcSkifQ==
</div>
Count the number of sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJudW1fc2VxID0gbGVuZ3RoKHByb3NlcSlcbnByaW50KHBhc3RlKFwiTnVtYmVyIG9mIHNlcXVlbmNlcyA9IFwiLCBudW1fc2VxKSkifQ==
</div>
<ul>
<li>Names of the sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJuYW1lcz1nZXROYW1lKHByb3NlcSlcbm5hbWVzICJ9
</div></li>
<li>Annotation of sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJhbm5vdGF0aW9uIDwtIGdldEFubm90KHByb3NlcSlcbmFubm90YXRpb24ifQ==
</div></li>
<li>Length of individual sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsZW5fc2VxID0gZ2V0TGVuZ3RoKHByb3NlcSlcbmxlbl9zZXEifQ==
</div></li>
<li>Protein statistics
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJwcmludChBQXN0YXQocHJvc2VxW1sxXV0pKSJ9
</div></li>
</ul>
It gives the type of proteins
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJmb3IgKGkgaW4gMTpudW1fc2VxKVxue1xuICBwcmludChuYW1lc1tpXSlcbiAgcHJpbnQoQUFzdGF0KHByb3NlcVtbaV1dKSlcbn0ifQ==
</div>
<ul>
<li>Show commands in seqinr
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsc2VxaW5yKCkifQ==
</div></li>
</ul>
</div>
<div id="using-short-read" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Using Short Read<a href="bioconductor.html#using-short-read" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This library can be installed by invoking the <code>BiocManager::install("ShortRead")</code> command to the console.</p>
load the library
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KFNob3J0UmVhZCkifQ==
</div>
<ul>
<li>Reading the fastq sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJmcSA9IHJlYWRGYXN0cShcImRhdGEvZmlsZTIuZmFzdHFcIilcbmZxIn0=
</div></li>
<li>Preview of first 3 sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJmcVsxOjNdIn0=
</div></li>
<li>Sequence information
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzcmVhZChmcSkifQ==
</div></li>
</ul>
Subset the sequence from second to fifth
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzcmVhZChmcVsyOjVdKSJ9
</div>
Inspect the quality of the sequence
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJxdWFsaXR5KGZxKSJ9
</div>
<ul>
<li>Number of sequence in the file
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsZW5ndGgoZnEpIn0=
</div></li>
<li>Length of reads
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJyZWFkTGVuZ3RoID0gd2lkdGgoZnEpXG5yZWFkTGVuZ3RoIn0=
</div></li>
<li>generating the summary of fasta sequences
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJmcV9yZWFkcyA9IHNyZWFkKGZxKVxuXG5pZF9vZl9yZWFkcz1pZChmcSlcbmlkX29mX3JlYWRzWzE6NV1cbmZxX3JlYWRzIn0=
</div></li>
<li>Seq length - we can generate the sequence lenght by converting it to a dataframe
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZXFfbGVuZ3RoID0gYXMuZGF0YS5mcmFtZShmcV9yZWFkc0ByYW5nZXNAd2lkdGgpXG5zZXFfbGVuZ3RoIn0=
</div></li>
<li>Quality scores
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJxdWFsaXR5X3Njb3Jlcz1xdWFsaXR5KGZxKVxucXVhbGl0eV9zY29yZXNbMTo1XSJ9
</div></li>
</ul>
Convert the scores to matrix and see as real numbers
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzY29yZXM9YXMocXVhbGl0eV9zY29yZXMsIFwibWF0cml4XCIpXG5zY29yZXMifQ==
</div>
<ul>
<li>Mean score for each sequence
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJtZWFuX3Njb3JlX2Zvcl9lYWNoX3NlcXVlbmNlID0gcm93TWVhbnMoc2NvcmVzLCBuYS5ybT1UUlVFKVxubWVhbl9zY29yZV9mb3JfZWFjaF9zZXF1ZW5jZSJ9
</div></li>
<li>Mean score for each position
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJtZWFuX3Njb3JlX2Zvcl9lYWNoX3Bvc2l0aW9uID0gcm93TWVhbnMoc2NvcmVzLCBuYS5ybT1UUlVFKVxubWVhbl9zY29yZV9mb3JfZWFjaF9wb3NpdGlvbiJ9
</div></li>
</ul>
Plot a boxplot for the scores
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJib3hwbG90KHNjb3JlcywgeGxhYj1cIlBvc2l0aW9uXCIsIHlsYWI9XCJQaHJlZCBzY29yZVwiKSJ9
</div>
</div>
<div id="genomic-ranges" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Genomic Ranges<a href="bioconductor.html#genomic-ranges" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>GenomicRanges</code> and <code>IRanges</code> packages are used in the calculations. These packages are masked in the biostrings.</p>
<ul>
<li><code>Genomic Ranges</code> are representing and manipulating genomic intervals and variables defined along a genome.</li>
<li><code>IRanges</code> used to provide an efficient handling of integer ranges, a fundamental data structure for representing genomic data.</li>
</ul>
Lets inspect a scenario where the libraries are used. The <code>IRanges</code> object will be randomly created.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KEdlbm9taWNSYW5nZXMpXG5saWJyYXJ5KElSYW5nZXMpXG5cbiMgQ3JlYXRpbmcgSVJhbmdlcyBvYmplY3RcbmlyIDwtIElSYW5nZXMoc3RhcnQ9YygxLDYsMTApLCBlbmQ9Yyg1LDgsMTIpKVxucHJpbnQoaXIpIn0=
</div>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KEdlbm9taWNSYW5nZXMpXG5saWJyYXJ5KElSYW5nZXMpXG5cbiMgQ3JlYXRpbmcgR2Vub21pY1JhbmdlcyBvYmplY3RcbmdyIDwtIEdSYW5nZXMoc2VxbmFtZXM9XCJjaHIxXCIsIHJhbmdlcz1JUmFuZ2VzKHN0YXJ0PWMoMSw2LDEwKSwgZW5kPWMoNSw4LDEyKSkpXG5wcmludChncikifQ==
</div>
<p>Genomic Ranges objects can be created from the IRanges. Lets now find the overlap in the genomic ranges created.</p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KEdlbm9taWNSYW5nZXMpXG5saWJyYXJ5KElSYW5nZXMpXG5cbiMgRmluZGluZyBvdmVybGFwc1xucXVlcnkgPC0gR1JhbmdlcyhzZXFuYW1lcz1cImNocjFcIiwgcmFuZ2VzPUlSYW5nZXMoc3RhcnQ9Yyg0LDcpLCBlbmQ9Yyg4LDEwKSkpXG5vdmVybGFwIDwtIGZpbmRPdmVybGFwcyhxdWVyeSwgZ3IpXG5wcmludChvdmVybGFwKSJ9
</div>
</div>
<div id="using-deseq-package" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Using `DESeq package<a href="bioconductor.html#using-deseq-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>DESeq</code> package is used to analyze the RNA-Seq data from the genome to find the gene expression. <code>EdgeR</code> can also be used especially where differential analysis of RNA data is required however in this section the <code>DESeq2</code> package will be used.</p>
Lets’ create a random data to work on the task
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KERFU2VxMilcblxuIyBMb2FkIGV4YW1wbGUgZGF0YVxuZGF0YSA8LSBtYXRyaXgocm5vcm0oMTAwMCksIG5jb2w9NSlcbmNvbERhdGEgPC0gZGF0YS5mcmFtZShjb25kaXRpb249cmVwKGMoXCJjb250cm9sXCIsIFwidHJlYXRlZFwiKSwgZWFjaD0yKSlcbmNvbERhdGEifQ==
</div>
Lets now use the randomly generated data to create DESeqDataSet
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KERFU2VxMilcblxuIyBMb2FkIGV4YW1wbGUgZGF0YVxuZGF0YSA8LSBtYXRyaXgocm5vcm0oMTAwMCksIG5jb2w9NSlcbmNvbERhdGEgPC0gZGF0YS5mcmFtZShjb25kaXRpb249cmVwKGMoXCJjb250cm9sXCIsIFwidHJlYXRlZFwiKSwgZWFjaD0yKSlcblxuIyBDcmVhdGUgREVTZXFEYXRhU2V0XG5kZHMgPC0gREVTZXFEYXRhU2V0RnJvbU1hdHJpeChjb3VudERhdGE9ZGF0YSwgY29sRGF0YT1jb2xEYXRhLCBkZXNpZ249fmNvbmRpdGlvbilcblxuIyBSdW4gREVTZXEgcGlwZWxpbmVcbmRkcyA8LSBERVNlcShkZHMpXG5yZXN1bHRzIDwtIHJlc3VsdHMoZGRzKVxuXG4jIFN1bW1hcml6ZSByZXN1bHRzXG5zdW1tYXJ5KHJlc3VsdHMpIn0=
</div>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJpZiAoIXJlcXVpcmVOYW1lc3BhY2UoXCJCaW9jTWFuYWdlclwiLCBxdWlldGx5ID0gVFJVRSkpXG4gICAgaW5zdGFsbC5wYWNrYWdlcyhcIkJpb2NNYW5hZ2VyXCIpXG5CaW9jTWFuYWdlcjo6aW5zdGFsbChcIkRFU2VxMlwiKVxuXG4jIENoZWNrIGRpbWVuc2lvbnMgb2YgZGF0YSBhbmQgY29sRGF0YVxucHJpbnQoZGltKGRhdGEpKVxucHJpbnQoZGltKGNvbERhdGEpKVxuXG4jIENoZWNrIGEgZmV3IHJvd3Mgb2YgY29sRGF0YVxucHJpbnQoaGVhZChjb2xEYXRhKSlcblxuIyBDaGVjayBpZiByb3cgbmFtZXMgYXJlIHNldCBjb3JyZWN0bHkgaW4gZGF0YVxucHJpbnQocm93bmFtZXMoZGF0YSkpXG5cbiMgVHJ5IHNldHRpbmcgcm93IG5hbWVzIGlmIHRoZXkgYXJlIG1pc3NpbmdcbmlmIChpcy5udWxsKHJvd25hbWVzKGRhdGEpKSkge1xuICByb3duYW1lcyhkYXRhKSA8LSBwYXN0ZTAoXCJHZW5lXCIsIHNlcV9sZW4obnJvdyhkYXRhKSkpXG59XG5cbiMgTG9hZCBERVNlcTIgbGlicmFyeVxubGlicmFyeShERVNlcTIpXG5cbiMgQ3JlYXRlIERFU2VxRGF0YVNldFxuZGRzIDwtIERFU2VxRGF0YVNldEZyb21NYXRyaXgoY291bnREYXRhID0gZGF0YSxcbiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIGNvbERhdGEgPSBjb2xEYXRhLFxuICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgZGVzaWduID0gfiBjb25kaXRpb24pIn0=
</div>
</div>
<div id="network-analysis-of-genomic-data" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> Network Analysis of Genomic Data<a href="bioconductor.html#network-analysis-of-genomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>“MarkovChain” is installed by <code>install.packages("markovchain")</code></p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIEV4YW1wbGUgb2YgY3JlYXRpbmcgYSBNYXJrb3YgY2hhaW4gb2JqZWN0IHdpdGggbWFya292Y2hhaW5cbmxpYnJhcnkobWFya292Y2hhaW4pXG5cbiMgRGVmaW5lIHN0YXRlcyBhbmQgdHJhbnNpdGlvbiBtYXRyaXggKHJlcGxhY2Ugd2l0aCBhY3R1YWwgZGF0YSlcbnN0YXRlcyA8LSBjKFwiVXAtcmVndWxhdGVkXCIsIFwiRG93bi1yZWd1bGF0ZWRcIiwgXCJObyBjaGFuZ2VcIilcbnRyYW5zaXRpb25fbWF0cml4IDwtIG1hdHJpeChjKDAuNywgMC4xLCAwLjIsXG4gICAgICAgICAgICAgICAgICAgICAgICAgICAgICAwLjIsIDAuNiwgMC4yLFxuICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgMC4xLCAwLjMsIDAuNiksIGJ5cm93ID0gVFJVRSwgbnJvdyA9IDMsIGRpbW5hbWVzID0gbGlzdChzdGF0ZXMsIHN0YXRlcykpXG5cbiMgQ3JlYXRlIGEgTWFya292IGNoYWluIG9iamVjdFxubWMgPC0gbmV3KFwibWFya292Y2hhaW5cIiwgc3RhdGVzID0gc3RhdGVzLCB0cmFuc2l0aW9uTWF0cml4ID0gdHJhbnNpdGlvbl9tYXRyaXgpXG5cbiMgRXhhbXBsZSBvZiBjb25zdHJ1Y3RpbmcgYW5kIGFuYWx5emluZyBhIGdlbmUgcmVndWxhdG9yeSBuZXR3b3JrIHdpdGggaWdyYXBoXG5saWJyYXJ5KGlncmFwaClcblxuIyBHZW5lcmF0ZSBleGFtcGxlIGRhdGEgKHJlcGxhY2Ugd2l0aCBhY3R1YWwgZGF0YSlcbmdlbmVzIDwtIGMoXCJHZW5lMVwiLCBcIkdlbmUyXCIsIFwiR2VuZTNcIiwgXCJHZW5lNFwiKVxuZWRnZXMgPC0gZGF0YS5mcmFtZShmcm9tID0gYyhcIkdlbmUxXCIsIFwiR2VuZTFcIiwgXCJHZW5lMlwiLCBcIkdlbmUzXCIpLFxuICAgICAgICAgICAgICAgICAgICB0byA9IGMoXCJHZW5lMlwiLCBcIkdlbmUzXCIsIFwiR2VuZTRcIiwgXCJHZW5lMVwiKSlcblxuIyBDcmVhdGUgYSBncmFwaCBvYmplY3RcbmcgPC0gZ3JhcGhfZnJvbV9kYXRhX2ZyYW1lKGVkZ2VzLCBkaXJlY3RlZCA9IFRSVUUsIHZlcnRpY2VzID0gZ2VuZXMpXG5cbiMgQW5hbHl6ZSB0aGUgZ3JhcGhcbnByaW50KHN1bW1hcnkoZykpXG5wbG90KGcsIGxheW91dCA9IGxheW91dC5jaXJjbGUpIn0=
</div>
<p>The generate data was fed into DESeq pipeline.</p>
</div>
<div id="code-challenge-6" class="section level2 hasAnchor" number="7.8">
<h2><span class="header-section-number">7.8</span> CODE CHALLENGE<a href="bioconductor.html#code-challenge-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this code challenge you will be required to installed the Bioconductor locally as guided above. Thereafter, you will install <code>MultiAssayExperiment</code> package that is part of the Bioconductor project. Below is the code to install the <code>MultiAssayExperiment</code> locally;</p>
<pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;MultiAssayExperiment&quot;)</code></pre>
<p>You will load the package along with the ggplot2 library and work on the <code>genCov</code> data. <code>genCov</code> data is an inbuilt data set from <code>MultiAssayExperiment</code> that contains simulated data showcasing the integration and analysis of multiple types of data from different assays. Challenge yourself with the following activities to get a clear understanding of genomic coverage data;</p>
<ol style="list-style-type: decimal">
<li>Load the <code>genCov</code> using the code, <code>data("genCov", package = "MultiAssayExperiment")</code>.<br />
</li>
<li>Check the structure <code>str()</code> of the data set and calculate the descriptive statistics.</li>
<li>Extract the coverage data with code <code>experiments(genCov)[["coverage"]]</code> and store in a variable <code>coverage_data</code>. Display the first 6 records of the the data.</li>
<li>Extract and inspect the genomic ranges from <code>genCov</code> with <code>rowRanges()</code> command. Display the first 6 genomic ranges.</li>
<li>Plot a line chart of the <code>"Coverage"</code> against the <code>"Genomic Position"</code>.</li>
<li>Calculate the average coverage for each sample in the <code>coverageData</code> using <code>colMeans()</code> command. Display the result on a bar chart.</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="ggplot.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/balsaedi/RCourse/edit/main/07-bioconductor_library_Advance-R.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/balsaedi/RCourse/blob/main/07-bioconductor_library_Advance-R.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
