<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Dpyr library | R Course</title>
  <meta name="description" content="Chapter 5 Dpyr library | R Course" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Dpyr library | R Course" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Dpyr library | R Course" />
  
  
  

<meta name="author" content="YOUR NAME HERE" />


<meta name="date" content="2024-05-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="working-with-files.html"/>
<link rel="next" href="ggplot.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SHORT TITLE HERE</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> R - Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#r---basic-concepts-and-applications"><i class="fa fa-check"></i><b>1.1</b> R - Basic Concepts and applications</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#why-r"><i class="fa fa-check"></i><b>1.2</b> Why R?</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#rstudio---look-and-feel"><i class="fa fa-check"></i><b>1.3.1</b> RStudio - Look and Feel</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#basic-mathematics-like-operators"><i class="fa fa-check"></i><b>1.4</b> Basic Mathematics like operators</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#basics"><i class="fa fa-check"></i><b>1.4.1</b> Basics</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#bodmas-rule"><i class="fa fa-check"></i><b>1.4.2</b> BODMAS rule</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#r-markdown"><i class="fa fa-check"></i><b>1.5</b> R Markdown</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#installation-of-packages"><i class="fa fa-check"></i><b>1.6</b> Installation of packages</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#assignment"><i class="fa fa-check"></i><b>1.7</b> Assignment</a></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#code-challenge"><i class="fa fa-check"></i><b>1.8</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html"><i class="fa fa-check"></i><b>2</b> Data Types and Data Structures</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-types"><i class="fa fa-check"></i><b>2.1</b> Data types</a></li>
<li class="chapter" data-level="2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-structures"><i class="fa fa-check"></i><b>2.2</b> Data Structures</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#vector"><i class="fa fa-check"></i><b>2.2.1</b> Vector</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#matrix"><i class="fa fa-check"></i><b>2.2.2</b> Matrix</a></li>
<li class="chapter" data-level="2.2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-frame"><i class="fa fa-check"></i><b>2.2.3</b> Data frame</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#code-challenge-1"><i class="fa fa-check"></i><b>2.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="loops.html"><a href="loops.html"><i class="fa fa-check"></i><b>3</b> Loops</a>
<ul>
<li class="chapter" data-level="3.1" data-path="loops.html"><a href="loops.html#for-loops"><i class="fa fa-check"></i><b>3.1</b> For loops</a></li>
<li class="chapter" data-level="3.2" data-path="loops.html"><a href="loops.html#while-loop"><i class="fa fa-check"></i><b>3.2</b> While loop</a></li>
<li class="chapter" data-level="3.3" data-path="loops.html"><a href="loops.html#code-challenge-2"><i class="fa fa-check"></i><b>3.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="working-with-files.html"><a href="working-with-files.html"><i class="fa fa-check"></i><b>4</b> Working with files</a>
<ul>
<li class="chapter" data-level="4.1" data-path="working-with-files.html"><a href="working-with-files.html#xlsx-xls-files"><i class="fa fa-check"></i><b>4.1</b> .XLSX, XLS files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="working-with-files.html"><a href="working-with-files.html#reading-a-file-to-a-dataframe"><i class="fa fa-check"></i><b>4.1.1</b> Reading a file to a dataframe</a></li>
<li class="chapter" data-level="4.1.2" data-path="working-with-files.html"><a href="working-with-files.html#write-a-new-table-to-a-file-locally"><i class="fa fa-check"></i><b>4.1.2</b> Write a new table to a file locally</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="working-with-files.html"><a href="working-with-files.html#csv-files"><i class="fa fa-check"></i><b>4.2</b> .CSV files</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="working-with-files.html"><a href="working-with-files.html#titanic-dataset"><i class="fa fa-check"></i><b>4.2.1</b> Titanic dataset</a></li>
<li class="chapter" data-level="4.2.2" data-path="working-with-files.html"><a href="working-with-files.html#ibm-employee-attrition-data"><i class="fa fa-check"></i><b>4.2.2</b> IBM Employee Attrition data</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="working-with-files.html"><a href="working-with-files.html#code-challenge-3"><i class="fa fa-check"></i><b>4.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dpyr-library.html"><a href="dpyr-library.html"><i class="fa fa-check"></i><b>5</b> Dpyr library</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dpyr-library.html"><a href="dpyr-library.html#dplyr-verbs"><i class="fa fa-check"></i><b>5.1</b> Dplyr Verbs</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="dpyr-library.html"><a href="dpyr-library.html#select"><i class="fa fa-check"></i><b>5.1.1</b> select</a></li>
<li class="chapter" data-level="5.1.2" data-path="dpyr-library.html"><a href="dpyr-library.html#filter"><i class="fa fa-check"></i><b>5.1.2</b> filter</a></li>
<li class="chapter" data-level="5.1.3" data-path="dpyr-library.html"><a href="dpyr-library.html#arrange"><i class="fa fa-check"></i><b>5.1.3</b> arrange</a></li>
<li class="chapter" data-level="5.1.4" data-path="dpyr-library.html"><a href="dpyr-library.html#mutate"><i class="fa fa-check"></i><b>5.1.4</b> mutate</a></li>
<li class="chapter" data-level="5.1.5" data-path="dpyr-library.html"><a href="dpyr-library.html#group_by"><i class="fa fa-check"></i><b>5.1.5</b> group_by</a></li>
<li class="chapter" data-level="5.1.6" data-path="dpyr-library.html"><a href="dpyr-library.html#summarise"><i class="fa fa-check"></i><b>5.1.6</b> Summarise</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="dpyr-library.html"><a href="dpyr-library.html#chaining"><i class="fa fa-check"></i><b>5.2</b> Chaining</a></li>
<li class="chapter" data-level="5.3" data-path="dpyr-library.html"><a href="dpyr-library.html#application-of-dplyr-on-microbiology"><i class="fa fa-check"></i><b>5.3</b> Application of Dplyr on Microbiology</a></li>
<li class="chapter" data-level="5.4" data-path="dpyr-library.html"><a href="dpyr-library.html#code-challenge-4"><i class="fa fa-check"></i><b>5.4</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ggplot.html"><a href="ggplot.html"><i class="fa fa-check"></i><b>6</b> GGPLOT</a>
<ul>
<li class="chapter" data-level="6.1" data-path="ggplot.html"><a href="ggplot.html#five-key-steps-to-ggplot"><i class="fa fa-check"></i><b>6.1</b> Five key steps to ggplot</a></li>
<li class="chapter" data-level="6.2" data-path="ggplot.html"><a href="ggplot.html#prepare-the-data-for-plotting---data-assessment"><i class="fa fa-check"></i><b>6.2</b> Prepare the data for plotting - Data Assessment</a></li>
<li class="chapter" data-level="6.3" data-path="ggplot.html"><a href="ggplot.html#plotting"><i class="fa fa-check"></i><b>6.3</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="ggplot.html"><a href="ggplot.html#base-set-up"><i class="fa fa-check"></i><b>6.3.1</b> Base set up</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="ggplot.html"><a href="ggplot.html#working-with-inbuilt-iris-data"><i class="fa fa-check"></i><b>6.4</b> Working with inbuilt iris data</a></li>
<li class="chapter" data-level="6.5" data-path="ggplot.html"><a href="ggplot.html#code-challenge-5"><i class="fa fa-check"></i><b>6.5</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>7</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="7.1" data-path="bioconductor.html"><a href="bioconductor.html#installing-biocmanager-package"><i class="fa fa-check"></i><b>7.1</b> Installing BiocManager Package</a></li>
<li class="chapter" data-level="7.2" data-path="bioconductor.html"><a href="bioconductor.html#searching-and-analyzing-the-strings"><i class="fa fa-check"></i><b>7.2</b> Searching and Analyzing the strings</a></li>
<li class="chapter" data-level="7.3" data-path="bioconductor.html"><a href="bioconductor.html#using-seqinr"><i class="fa fa-check"></i><b>7.3</b> Using seqinr</a></li>
<li class="chapter" data-level="7.4" data-path="bioconductor.html"><a href="bioconductor.html#using-short-read"><i class="fa fa-check"></i><b>7.4</b> Using Short Read</a></li>
<li class="chapter" data-level="7.5" data-path="bioconductor.html"><a href="bioconductor.html#code-challenge-6"><i class="fa fa-check"></i><b>7.5</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dpyr-library" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Dpyr library<a href="dpyr-library.html#dpyr-library" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script src="js/dcl.js"></script>
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
<script src="js/dcl.js"></script>
<p><code>readxl</code> library has been used to read excel table into R and manipulate the data using R basic functions, now <code>dplyr</code> library will be used. This, <code>dplyr</code> library is an advanced package that has functions used to manipulate data. The functions are referred to as dplyr verbs. This are some of the examples of dplyr verbs;</p>
<ul>
<li>select</li>
<li>filter</li>
<li>arrange</li>
<li>mutate</li>
<li>group by</li>
<li>summarise</li>
</ul>
<p>To install <code>dplyr</code>, run the below command on the console.</p>
<pre><code>install.packages(&quot;dplyr&quot;)</code></pre>
<p>Ensure your computer is connected to the internet when installing!</p>
To invoke the libraries to the program run;
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJsaWJyYXJ5KGRwbHlyKVxubGlicmFyeShyZWFkeGwpIn0=
</div>
<p>Before working on the data, it need to imported first. The <code>readxl</code> library will be used. The data will then be manipulated by <code>dplyr</code>.</p>
Apart from data retrieved from excel tables, there are some inbuilt freely available datasets in R that a student can use for practice. <code>data()</code> function can be used to view some of the freely available datasets in R.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJkYXRhKCkifQ==
</div>
This will show all data sets including both the inbuilt and the previously imported ones. Alternatively the code below will show only the inbuilt R data sets.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJscyhcInBhY2thZ2U6ZGF0YXNldHNcIikifQ==
</div>
Some libraries come with their own inbuilt data sets which can be explored by the command;
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJkYXRhKHBhY2thZ2U9LnBhY2thZ2VzKGFsbC5hdmFpbGFibGUgPSBUUlVFKSkifQ==
</div>
<p>Good example is the famous iris that will be used in this course. It is always used by beginners in data science. The data set contains measurements of various characteristics of iris flowers. These characteristics include sepal length, sepal width, petal length, and petal width. There are three species of iris flowers in the data set: setosa, versicolor, and virginica.
The data will be invoked to R before assessment and wrangling;</p>
Lets load the iris data and explore the first few records.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJkYXRhKFwiaXJpc1wiKVxuaGVhZChpcmlzKSJ9
</div>
Calculating the summary statistics
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzdW1tYXJ5KGlyaXMpIn0=
</div>
Lets write this data set locally to a file.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJ3cml0ZS50YWJsZShpcmlzLCBcImRhdGEvaXJpc19kYXRhc2V0LnR4dFwiLCBhcHBlbmQgPSBGQUxTRSlcbmZpbGUuc2hvdyhcImRhdGEvaXJpc19kYXRhc2V0LnR4dFwiKSJ9
</div>
or to remove quotes next to values
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJ3cml0ZS50YWJsZShpcmlzLCBcImRhdGEvaXJpc19kYXRhc2V0LnR4dFwiLCBhcHBlbmQgPSBGQUxTRSwgcXVvdGUgPSBGQUxTRSlcbmZpbGUuc2hvdyhcImRhdGEvaXJpc19kYXRhc2V0LnR4dFwiKSJ9
</div>
or to remove quotes and row names
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJ3cml0ZS50YWJsZShpcmlzLCBcImRhdGEvaXJpc19kYXRhc2V0LnR4dFwiLCBhcHBlbmQgPSBGQUxTRSwgcXVvdGU9RkFMU0UsIHJvdy5uYW1lcyA9IEZBTFNFKVxuZmlsZS5zaG93KFwiZGF0YS9pcmlzX2RhdGFzZXQudHh0XCIpIn0=
</div>
<div id="dplyr-verbs" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Dplyr Verbs<a href="dpyr-library.html#dplyr-verbs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="select" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> select<a href="dpyr-library.html#select" class="anchor-section" aria-label="Anchor link to header"></a></h3>
This dplyr verb is used when selecting or dropping specific columns. In this lesson we will find the iris column names and select two of them using <code>select</code>.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIEZpbmQgdGhlIGNvbHVtbiBuYW1lcyBcbmNvbG5hbWVzKGlyaXMpIn0=
</div>
<p>Remember the data frame to work on need to be specified in the arguments such that</p>
<pre><code>selected_data = select(data_frame, col1, col2, col3)</code></pre>
<p>Therefore, we will select the columns; Species, Petal length and petal width.</p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZWxlY3RlZF9pcmlzX2RhdGEgPSBzZWxlY3QoaXJpcyxcbiAgICAgICAgICAgICAgICAgICAgICAgUGV0YWwuTGVuZ3RoLCBQZXRhbC5XaWR0aCwgU3BlY2llcylcblxuIyB2aWV3IHRoZSBmaXJzdCBmZXcgcm93cyBvZiB0aGUgc2VsZWN0ZWQgZGF0YVxuaGVhZChzZWxlY3RlZF9pcmlzX2RhdGEpIn0=
</div>
<p>The three selected columns are displayed in the data frame above.</p>
<p>Specific columns can be dropped by putting <code>-</code> before the column name as</p>
<pre><code># Drop specified columns 
remaining_data = select(data_frame, -col1_to_drop, -col2_to_drop)</code></pre>
In this lesson, we will drop petal length, petal width and Species columns;
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJyZW1haW5pbmdfaXJpc19kYXRhID0gc2VsZWN0KGlyaXMsXG4gICAgICAgICAgICAgICAgICAgICAgIC1QZXRhbC5MZW5ndGgsIC1QZXRhbC5XaWR0aCwgLVNwZWNpZXMpXG5cbiMgdmlldyB0aGUgZmlyc3QgZmV3IHJvd3Mgb2YgdGhlIHNlbGVjdGVkIGRhdGFcbmhlYWQocmVtYWluaW5nX2lyaXNfZGF0YSkifQ==
</div>
<p><span style="color: green;"><strong>Activity</strong></span></p>
<p>You will be required to use the <code>car_sales</code> data set from <a href="https://raw.githubusercontent.com/insaid2018/Term-1/master/Data/Projects/car_sales.csv.Read" class="uri">https://raw.githubusercontent.com/insaid2018/Term-1/master/Data/Projects/car_sales.csv.Read</a> the data using <code>read.csv</code> and select the <code>car, price, body, mileage, engV, engType, year, model</code>. Save the data frame from the selected columns as <code>selected_cars_df</code>. Show the first few rows of the <code>selected_cars_df</code>.</p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIExvYWQgdGhlIGRwbHlyIGxpYnJhcnlcbiMgQ09ERSBIRVJFXG5cbiMgUmVhZCB0aGUgZGF0YSBcbiMgQ09ERSBIRVJFXG5cbiMgc2VsZWN0IHRoZSByZXF1aXJlZCBjb2x1bW5zIFxuIyBDT0RFIEhFUkVcblxuIyBTaG93IHRoZSBmaXJzdCBmZXcgcm93cyBvZiB0aGUgZGF0YSBmcmFtZVxuIyBDT0RFIEhFUkUiLCJzb2x1dGlvbiI6IiMgTG9hZCB0aGUgZHBseXIgbGlicmFyeVxubGlicmFyeShkcGx5cilcblxuIyBSZWFkIHRoZSBkYXRhIFxuZGF0YV9saW5rID0gXCJodHRwczovL3Jhdy5naXRodWJ1c2VyY29udGVudC5jb20vaW5zYWlkMjAxOC9UZXJtLTEvbWFzdGVyL0RhdGEvUHJvamVjdHMvY2FyX3NhbGVzLmNzdlwiXG5jYXJfc2FsZXMgPSByZWFkLmNzdihkYXRhX2xpbmspXG5cbiMgc2VsZWN0IHRoZSByZXF1aXJlZCBjb2x1bW5zIFxuc2VsZWN0ZWRfY2Fyc19kZiA9IHNlbGVjdChjYXJfc2FsZXMsIFxuICAgICAgICAgICAgICAgICAgICAgICAgICBjYXIsIHByaWNlLCBib2R5LCBtaWxlYWdlLCBlbmdWLCBlbmdUeXBlLCB5ZWFyLCBtb2RlbClcblxuIyBTaG93IHRoZSBmaXJzdCBmZXcgcm93cyBvZiB0aGUgZGF0YSBmcmFtZSBcbmhlYWQoc2VsZWN0ZWRfY2Fyc19kZikifQ==
</div>
</div>
<div id="filter" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> filter<a href="dpyr-library.html#filter" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Is a verb/function from <code>dplyr</code> used to filter records in a data frame based on a specific condition. It allows the analyst to retrieve the records he/she is interested in and work easier with the subset.
With <code>filter()</code>, the data frame and the condion are passed as a arguments;</p>
<pre><code># Filtering rows where a certain column meets a condition
filtered_data = filter(data_frame, 
column_name &gt; 5 # This is the condition)</code></pre>
Lets select the species ‘setosa’ from the iris data set
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzZXRvc2FfaXJpcyA9IGZpbHRlcihpcmlzLCAjIHRoZSBkYXRhIGZyYW1lXG4gICAgICAgICAgICAgICAgICAgICBTcGVjaWVzID09IFwic2V0b3NhXCIgIyB0aGUgY29uZGl0aW9uXG4gICAgICAgICAgICAgICAgICAgICApXG4jIEZpcnN0IGZldyByZWNvcmRzIG9mIHNldG9zYSBkYXRhXG5oZWFkKHNldG9zYV9pcmlzKSJ9
</div>
Records with sepal width of more than 3.0 can be filtered. Here is how we achieve such a subset
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJ3aWRlX3NlcGFsX2lyaXMgPSBmaWx0ZXIoaXJpcywgI3RoZSBkYXRhIGZyYW1lXG4gICAgICAgICAgICAgICAgICAgICAgICAgU2VwYWwuV2lkdGg+My4wICMgdGhlIGNvbmRpdGlvblxuICAgICAgICAgICAgICAgICAgICAgICAgIClcblxuaGVhZCh3aWRlX3NlcGFsX2lyaXMpIn0=
</div>
<p><span style="color: green;"><strong>Activity</strong></span></p>
<p>With the <code>car_sales</code> data set that you used above, use <code>filter()</code> function to get the cars that were sold from the year 2008 to present and name them <code>latest_car_sales</code>. Count the number of observations made and show the first few rows.</p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIExvYWQgdGhlIGRwbHlyIGxpYnJhcnlcbmxpYnJhcnkoZHBseXIpXG5cbiMgUmVhZCB0aGUgZGF0YSBcbmRhdGFfbGluayA9IFwiaHR0cHM6Ly9yYXcuZ2l0aHVidXNlcmNvbnRlbnQuY29tL2luc2FpZDIwMTgvVGVybS0xL21hc3Rlci9EYXRhL1Byb2plY3RzL2Nhcl9zYWxlcy5jc3ZcIlxuY2FyX3NhbGVzID0gcmVhZC5jc3YoZGF0YV9saW5rKVxuXG4jIEZpbHRlciB0byBmaW5kIGNhcnMgc29sZCBmcm9tIDIwMDhcbmxhdGVzdF9jYXJfc2FsZXMgPSBmaWx0ZXIoY2FyX3NhbGVzLCBcbiAgICAgICAgICAgICAgICAgICAgICAgICAgeWVhcj49MjAwOClcblxuIyBDb3VudCB0aGUgb2JzZXJ2YXRpb25zIG1hZGUuIFVzZSBucm93cyBmdW5jdGlvblxubnJvd3MobGF0ZXN0X2Nhcl9zYWxlcylcblxuIyBTaG93IHRoZSBmaXJzdCBmZXcgcm93c1xuaGVhZChsYXRlc3RfY2FyX3NhbGVzKSJ9
</div>
</div>
<div id="arrange" class="section level3 hasAnchor" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> arrange<a href="dpyr-library.html#arrange" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is <code>dplyr</code> verb/function used for sorting rows by rearranging in a specific order. here is how to use <code>arrange()</code> function;</p>
<pre><code>arranged_data = arrange(data_frame, column_name)</code></pre>
<p>This allows the analyst to arrange the data in a default ascending order. To arrange in a descending order a <code>desc()</code> function is added as;</p>
<pre><code># Note the additional desc function
arranged_data = arrange(data_frame, desc(column_name))</code></pre>
Now lets order the iris data in an ascending order based on Petal length and view the first 6 records with the shortest petal.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIFNvcnQgdGhlIGRhdGEgXG5ieV9wZXRhbF9sZW5ndGggPSBhcnJhbmdlKGlyaXMsICMgZGF0YSBmcmFtZSBcbiAgICAgICAgICAgICAgICAgICAgICAgICAgUGV0YWwuTGVuZ3RoICMgb3JkZXIgYnkgY29sdW1uXG4gICAgICAgICAgICAgICAgICAgICAgICAgIClcblxuIyBWaWV3IHRoZSBkYXRhIFxuaGVhZChieV9wZXRhbF9sZW5ndGgpIn0=
</div>
Lets repeat the same process but now we order the data in a descending order.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIFNvcnQgdGhlIGRhdGEgXG5ieV9wZXRhbF9sZW5ndGggPSBhcnJhbmdlKGlyaXMsICMgZGF0YSBmcmFtZSBcbiAgICAgICAgICAgICAgICAgICAgICAgICAgZGVzYyhQZXRhbC5MZW5ndGgpICMgb3JkZXIgYnkgY29sdW1uXG4gICAgICAgICAgICAgICAgICAgICAgICAgIClcblxuIyBWaWV3IHRoZSBkYXRhIFxuaGVhZChieV9wZXRhbF9sZW5ndGgpIn0=
</div>
<p><span style="color: green;"><strong>Activity</strong></span>
Arrange the columns in the <code>car_sales</code> data set according to <code>mileage</code> in descending order. Show the last few rows</p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIExvYWQgdGhlIGRwbHlyIGxpYnJhcnlcbiMgQ09ERSBIRVJFIFxuXG4jIFJlYWQgdGhlIGRhdGEgaW50byBhIGRhdGEgZnJhbWVcbiMgQ09ERSBIRVJFIFxuXG4jIE9yZGVyIGFjY29yZGluZyB0byBtaWxlYWdlIGluIGRlc2NlbmRpbmcgb3JkZXJcbiMgQ09ERSBIRVJFIFxuXG4jIFNob3cgdGhlIGxhc3QgZmV3IHJvd3Mgb2YgdGhlIGRhdGEgc2V0XG4jIENPREUgSEVSRSIsInNvbHV0aW9uIjoiIyBMb2FkIHRoZSBkcGx5ciBsaWJyYXJ5XG5saWJyYXJ5KGRwbHlyKVxuXG4jIFJlYWQgdGhlIGRhdGEgaW50byBhIGRhdGEgZnJhbWVcbmRhdGFfbGluayA9IFwiaHR0cHM6Ly9yYXcuZ2l0aHVidXNlcmNvbnRlbnQuY29tL2luc2FpZDIwMTgvVGVybS0xL21hc3Rlci9EYXRhL1Byb2plY3RzL2Nhcl9zYWxlcy5jc3ZcIlxuY2FyX3NhbGVzID0gcmVhZC5jc3YoZGF0YV9saW5rKVxuXG4jIE9yZGVyIGFjY29yZGluZyB0byBtaWxlYWdlIGluIGRlc2NlbmRpbmcgb3JkZXJcbm1pbGVhZ2Vfb3JkZXIgPSBhcnJhbmdlKGNhcl9zYWxlcywgXG4gICAgICAgICAgICAgICAgICAgICAgICBkZXNjKG1pbGVhZ2UpKVxuXG4jIFNob3cgdGhlIGxhc3QgZmV3IHJvd3Mgb2YgdGhlIGRhdGEgc2V0XG50YWlsKG1pbGVhZ2Vfb3JkZXIpIn0=
</div>
</div>
<div id="mutate" class="section level3 hasAnchor" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> mutate<a href="dpyr-library.html#mutate" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<code>mutate()</code> is a dplyr verb used to modifying the existing variables or creating new variables in a data set.
In this case we can calculate the log off Sepal length in the iris data
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiIjIG1vZGlmeSBTZXBhbC5MZW5ndGhcbm5ld19pcmlzID0gbXV0YXRlKGlyaXMsIFNlcGFsLkxlbmd0aD1sb2coU2VwYWwuTGVuZ3RoKSlcblxuaGVhZChuZXdfaXJpcykifQ==
</div>
Additionally, we can create an entirely new variable by <code>mutate()</code>. In this case we will find the ratio between petal length and petal width. The new variable will be called “Petal.Length.Width.Ratio”
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJuZXdfaXJpcyA9IG11dGF0ZShpcmlzLFxuICAgICAgICAgICAgICAgICAgUGV0YWwuTGVuZ3RoLldpZHRoLlJhdGlvID0gUGV0YWwuTGVuZ3RoL1BldGFsLldpZHRoKVxuXG5oZWFkKG5ld19pcmlzKSJ9
</div>
<p>The “Petal.Length.Width.Ratio” is found by dividing the Petal.Length and the Petal.Width variables.</p>
</div>
<div id="group_by" class="section level3 hasAnchor" number="5.1.5">
<h3><span class="header-section-number">5.1.5</span> group_by<a href="dpyr-library.html#group_by" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>group_by()</code> is a function used to group records in a data frame by one or more variables. It allows the analyst to create a group based on a certain criteria.</p>
Lets group the iris data based on the Species variable;
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJpcmlzX2dyb3VwcyA9IGdyb3VwX2J5KGlyaXMsIFNwZWNpZXMpXG5oZWFkKGlyaXNfZ3JvdXBzKSJ9
</div>
<p>This groupings allow the analyst to retrieve insights at more base level and uncover more insights that could not have been possible when analyzing the entire data set</p>
</div>
<div id="summarise" class="section level3 hasAnchor" number="5.1.6">
<h3><span class="header-section-number">5.1.6</span> Summarise<a href="dpyr-library.html#summarise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
To calculate summary statistics such as average, median and maximum the <code>summarise()</code> is used. This function collapses mutilple rows into a summary row. For instance calculating the mean Petal width;
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJzdW1tYXJpc2UoaXJpcywgXG4gICAgICAgICAgbWVhbl9wZXRhbF93aWR0aD1tZWFuKFBldGFsLldpZHRoKSkifQ==
</div>
To find the mean petal width for each iris species;
- the iris data will be grouped by species
- a mean value for each group will be calculated
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJncm91cGVkX2lyaXMgPSBncm91cF9ieShpcmlzLCBTcGVjaWVzKVxubWVhbl9wZXRhbF93aWR0aHMgPSBzdW1tYXJpc2UoZ3JvdXBlZF9pcmlzLCBtZWFuX3ZhbHVlPW1lYW4oUGV0YWwuV2lkdGgpKVxubWVhbl9wZXRhbF93aWR0aHMifQ==
</div>
</div>
</div>
<div id="chaining" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Chaining<a href="dpyr-library.html#chaining" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Chaining is the process of combining several operations together using the <code>%&gt;%</code> or forward pipe operator. The chained workflow succeeds each other until the whole process is done. To understand chaining, the mtcars(Motor Trend cars) data set will be used. Mtcars is also a well-known data set containing several attributes of 32 different cars from 1974. Here’s a brief explanation of the variables in the mtcars data set:</p>
<ol style="list-style-type: decimal">
<li>mpg: Miles per gallon (fuel efficiency).</li>
<li>cyl: Number of cylinders.</li>
<li>disp: Displacement (cubic inches).</li>
<li>hp: Horsepower.</li>
<li>drat: Rear axle ratio.</li>
<li>wt: Weight (in 1000 lbs).</li>
<li>qsec: Quarter mile time (in seconds).</li>
<li>vs: Engine type (0 = V-shaped, 1 = Straight).</li>
<li>am: Transmission type (0 = Automatic, 1 = Manual).</li>
<li>gear: Number of forward gears.</li>
<li>carb: Number of carburetors.</li>
</ol>
Lets start by loading the data into the program and view its first few records;
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJkYXRhKG10Y2FycylcbmhlYWQobXRjYXJzKSJ9
</div>
Lets <code>select</code> 6 most important columns in this analysis
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJjYXJzMSA9IG10Y2FycyAlPiUgc2VsZWN0KG1wZywgY3lsLCBkaXNwLCBocCwgcXNlYywgYW0pXG5cbmhlYWQoY2FyczEpIn0=
</div>
Lets now <code>filter</code> to find vehicles with an automatic transmission type. The <code>filter</code> verb will be chained to <code>select</code> verb with <code>%&gt;%</code>.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJjYXJzMiA9IG10Y2FycyAlPiVzZWxlY3QobXBnLCBjeWwsIGRpc3AsIGhwLCBxc2VjLCBhbSkgJT4lXG4gIGZpbHRlcihhbT09MClcbmhlYWQoY2FyczIpIn0=
</div>
All these vehicles are of automatic transmission type, lets rank them according to the horsepower in descending order.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJjYXJzMz0gbXRjYXJzICU+JXNlbGVjdChtcGcsIGN5bCwgZGlzcCwgaHAsIHFzZWMsIGFtLCB3dCkgJT4lXG4gIGZpbHRlcihhbT09MCkgJT4lXG4gIGFycmFuZ2UoZGVzYyhocCkpXG5cbmhlYWQoY2FyczMpIn0=
</div>
A new column of weight in 1000kgs (wt_1000kgs) can be created by diving weight in 1000lbs by 2.20462. <code>mutate</code> verb will be chained also.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJjYXJzND0gbXRjYXJzICU+JXNlbGVjdChtcGcsIGN5bCwgZGlzcCwgaHAsIHFzZWMsIGFtLCB3dCkgJT4lXG4gIGZpbHRlcihhbT09MCkgJT4lXG4gIGFycmFuZ2UoZGVzYyhocCkpICU+JVxuICBtdXRhdGUod3RfMTAwMGtncz13dC8yLjIwNDYyKVxuXG5oZWFkKGNhcnM0KSJ9
</div>
<p>The above process has explained how chained works in <code>dplyr</code>. Many functions/processed can be chained together to manipulate data to the desired output. The next section will apply chaining to biology and be used to answer a few questions that will cement your understanding in R as a biologist.</p>
</div>
<div id="application-of-dplyr-on-microbiology" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Application of Dplyr on Microbiology<a href="dpyr-library.html#application-of-dplyr-on-microbiology" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Genetics composition of an organism can be analyzed using <code>dplyr</code>, in this course unit it will be used to analyze the human genetics data.</p>
<ul>
<li>Read the data set from a file into a data frame
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZyA9IHJlYWRfZXhjZWwoXCJkYXRhL0h1bWFuX2dlbm9tZS54bHN4XCIpXG5jbGFzcyhoZykifQ==
</div></li>
</ul>
review the first six records of the data set
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZWFkKGhnKSJ9
</div>
the last six records of the data set
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJ0YWlsKGhnKSJ9
</div>
<code>sample_n()</code> command is used to randomly sample fixed number of rows from the data frame. Lets sample five rows randomly
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZyAlPiUgc2FtcGxlX24oNSkifQ==
</div>
<code>sample_fra()</code> command can also be used to extract a percent of rows randomly. Lets samople 25% of the rows
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZyAlPiUgc2FtcGxlX2ZyYWMoMC4yNSwgcmVwbGFjZSA9IEZBTFNFKSJ9
</div>
<p>The <code>replace</code> argument allows repetition of rows when sampling when set <code>TRUE</code> and vice versa. The essence of <code>dplyr</code> in microbiology data can be explained better by questions.</p>
<p><strong>Q1. Which chromosomes has maximum size</strong></p>
<p>The functions used here are; select, arrange, slice</p>
<ol style="list-style-type: decimal">
<li>The required columns will be selected i.e Chromosomes and Base_Pairs(size)</li>
<li>The data will ordered according to Base_Pairs in descending order by the <code>arrange</code> command.<br />
</li>
<li>the data will be “sliced” to select the first row</li>
</ol>
<p>All these processes will be chained to run progressively from process 1 to 3.</p>
All the above steps should be piped
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZzEgPSBoZyAlPiUgI3RyYW5zZmVyIHRoZSBkYXRhIHRvIGEgbmV3IHZhcmlhYmxlXG4gIHNlbGVjdChDaHJvbW9zb21lLCBCYXNlX1BhaXJzKSAlPiUgIyBzZWxlY3QgdGhlIHJlcXVpcmVkIGNvbHVtbnMgXG4gICAgICAgICAgIGFycmFuZ2UoZGVzYyhCYXNlX1BhaXJzKSkgJT4lICNvcmRlciBpbiBkZXNjZW5kaW5nIG9yZGVyXG4gICAgICAgICAgIHNsaWNlKDEpXG5cbmhnMSJ9
</div>
<p><strong>Q2. List of Chromosomes with more than 500 pseudogenes and more than 1000 protein coding genes</strong></p>
<p>The human genome data frame will be filtered to find the Chromosomes with more than 500 pseudogenes and more than 1000 protein coding genes.
<code>&amp;</code> condition is used in the filter</p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZzI9aGclPiVcbiAgZmlsdGVyKFBzZXVkb2dlbmVzPjUwMCAmIFByb3RlaW5fQ29kaW5nX2dlbmVzPjEwMDApXG5oZzIifQ==
</div>
<p><strong>Q3. List of Chromosomes with more tha 500 pseudogenes or more than 1000 protein coding genes and miRNA more than 100</strong></p>
<p>a frame will be filtered to find the Chromosomes with more than 500 pseudogenes
The human genome dator more than 1000 protein coding genes and mtRNA more than 100.</p>
The <code>&amp;</code> and <code>|</code> conditions will be used in place of “and” and “or” respectively.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZzI9aGcgJT4lIFxuICBmaWx0ZXIoKFBzZXVkb2dlbmVzPjUwMHxQcm90ZWluX0NvZGluZ19nZW5lcz4xMDAwKSAmIG1pUk5BPjEwMClcbmhnMiJ9
</div>
<p><strong>Q4. Which 10 Chromosomes have the highest protein gene density (Genes per megabase)</strong></p>
<ul>
<li>Mutate -&gt; is a command to create a new column</li>
</ul>
Lets create a column of genes per megabase where a megabase is 1000000 base pairs
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZzM9aGcgJT4lXG4gIG11dGF0ZShHZW5lc19wZXJfTWI9cm91bmQoKFByb3RlaW5fQ29kaW5nX2dlbmVzLyhCYXNlX1BhaXJzLzEwMDAwMCkpLFxuICAgICAgICAgICAgICAgICAgICAgICAgICAgIGRpZ2l0PTIpKVxuaGczIn0=
</div>
then order the data according to genes per megabse column (<code>Genes_per_mb</code>) in descending order and select the first 10 rows using <code>arrange</code> and <code>select</code> commands.
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZzQ9aGczICU+JVxuICBhcnJhbmdlKGRlc2MoR2VuZXNfcGVyX01iKSkgJT4lICMgb3JkZXIgaW4gZGVzY2VuZGluZyBvcmRlciBcbiAgc2VsZWN0KENocm9tb3NvbWUsIEdlbmVzX3Blcl9NYikgJT4lICMgc2VsZWN0IHRoZSByZXF1aXJlZCBjb2x1bW5zIFxuICBzbGljZSgxOjEwKSAjIHNlbGVjdCB0aGUgdG9wIDEwXG5cbmhnNCJ9
</div>
<p><strong>Q5. Calculate the mean of gene density for autosomes, sex chromosomes and mitochondrial genes</strong></p>
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJhdXRvc29tZXM9cmVwKFwiQVwiLCAyMikgI2NyZWF0ZSBhIHZlY3RvciB3aXRoIDIyIEFzXG5wcmludChhdXRvc29tZXMpXG5jaHJfY2xhc3M9YyhhdXRvc29tZXMsIFwiU1wiLCBcIlNcIiwgXCJNXCIpXG5jaHJfY2xhc3NcbmhnNT0gaGczICU+JVxuICBtdXRhdGUoVHlwZT1jaHJfY2xhc3MpXG5cbmhnNSAjIGNvbmZpcm0gdGhlIG5ldyBjb2x1bW4gXCJUeXBlXCIgaXMgYWRkZWQifQ==
</div>
Group by chromosomes and calculate the mean density
<div data-datacamp-exercise="" data-height="300" data-encoded="true">
eyJsYW5ndWFnZSI6InIiLCJzYW1wbGUiOiJoZzY9aGc1ICU+JSBcbiAgZ3JvdXBfYnkoVHlwZSkgJT4lXG4gIHN1bW1hcmlzZShnZW5lX2RlbnNpdHk9bWVhbihHZW5lc19wZXJfTWIpKVxuaGc2In0=
</div>
</div>
<div id="code-challenge-4" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> CODE CHALLENGE<a href="dpyr-library.html#code-challenge-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><code>ToothGrowth</code> is one of the freely available data sets in base R. Go on and;
<ol style="list-style-type: lower-alpha">
<li>Load the <code>ToothGrowth</code> data set.</li>
<li>Display the first 5 records.</li>
<li>Use the function <code>write.table</code> to write the data set to a local file.</li>
</ol></li>
<li>You will use the Palmer Archipelago(Antarctica) Penguin data set from kaggle to practice various data manipulation techniques using <code>dplyr</code> verbs in R. Ensure You load the data and necessary packages before working on the following questions below;
<ol style="list-style-type: lower-alpha">
<li>Using <code>select()</code> verb, create a new data set that contains only the columns <code>species, island, bill_length_mm</code>.</li>
<li>Filter the Penguin data set to retain only the “Adelie” species that live in “Torgersen” island. Use <code>filter()</code> function.</li>
<li>Arrange the Penguin data set according to <code>bill_lenghth_mm</code> in descending order. Use <code>arrange()</code> verb.</li>
<li>Using the <code>mutate()</code> function, add a new column called <code>bill_length_cm</code> derived from <code>bill_length_mm</code> by 10.</li>
<li>Group the data set according to <code>island</code>.</li>
<li>For each species and island, calculate the mean <code>bill_length_mm</code> and the observations count.</li>
</ol></li>
<li>In this activity, you will chain at least four <code>dplyr</code> verbs to achieve the desired outcome.
<ul>
<li>Select the columns, <code>island, body_mass_g, species, bill_length_mm</code></li>
<li>Filter the selected data set to include penguins that have a bill length greater than 40mm</li>
<li>Create a new column/variable that will convert <code>bill_length_mm</code> to <code>bill_length_cm</code> i.e divide <code>bill_length_mm</code> by 10 to come up with <code>bill_length_cm</code>.</li>
<li>Order the resulting data set by <code>body_mass_g</code> in descending order.</li>
</ul></li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="working-with-files.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="ggplot.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/balsaedi/RCourse/edit/main/05-dplyr_library-advance-R.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/balsaedi/RCourse/blob/main/05-dplyr_library-advance-R.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
