<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 23 Video 8 | YOUR TITLE HERE</title>
  <meta name="description" content="Chapter 23 Video 8 | YOUR TITLE HERE" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 23 Video 8 | YOUR TITLE HERE" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 23 Video 8 | YOUR TITLE HERE" />
  
  
  

<meta name="author" content="YOUR NAME HERE" />


<meta name="date" content="2024-04-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="r-markdown.html"/>
<link rel="next" href="using-seqinr.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SHORT TITLE HERE</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Down the rabbit-hole</a></li>
<li class="chapter" data-level="2" data-path="r---basic-concepts-and-applications.html"><a href="r---basic-concepts-and-applications.html"><i class="fa fa-check"></i><b>2</b> R - Basic Concepts and applications</a></li>
<li class="chapter" data-level="3" data-path="why-r.html"><a href="why-r.html"><i class="fa fa-check"></i><b>3</b> Why R?</a></li>
<li class="chapter" data-level="4" data-path="r---look-and-feel.html"><a href="r---look-and-feel.html"><i class="fa fa-check"></i><b>4</b> R - Look and Feel</a></li>
<li class="chapter" data-level="5" data-path="r-studio.html"><a href="r-studio.html"><i class="fa fa-check"></i><b>5</b> R studio</a>
<ul>
<li class="chapter" data-level="5.1" data-path="r-studio.html"><a href="r-studio.html#data-types"><i class="fa fa-check"></i><b>5.1</b> Data types</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-structure.html"><a href="data-structure.html"><i class="fa fa-check"></i><b>6</b> Data Structure</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-structure.html"><a href="data-structure.html#vector"><i class="fa fa-check"></i><b>6.1</b> Vector</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="video-2.html"><a href="video-2.html"><i class="fa fa-check"></i><b>7</b> Video 2</a></li>
<li class="chapter" data-level="8" data-path="matrix.html"><a href="matrix.html"><i class="fa fa-check"></i><b>8</b> Matrix</a></li>
<li class="chapter" data-level="9" data-path="video-3.html"><a href="video-3.html"><i class="fa fa-check"></i><b>9</b> Video 3</a></li>
<li class="chapter" data-level="10" data-path="dataframes.html"><a href="dataframes.html"><i class="fa fa-check"></i><b>10</b> Dataframes</a></li>
<li class="chapter" data-level="11" data-path="rloops.html"><a href="rloops.html"><i class="fa fa-check"></i><b>11</b> Rloops</a>
<ul>
<li class="chapter" data-level="11.1" data-path="rloops.html"><a href="rloops.html#for-loops"><i class="fa fa-check"></i><b>11.1</b> for loops</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="q1.-calculate-the-gene-density---genesmegabase-for-each-chromosome.html"><a href="q1.-calculate-the-gene-density---genesmegabase-for-each-chromosome.html"><i class="fa fa-check"></i><b>12</b> Q1. Calculate the gene density - genes/megabase for each chromosome</a></li>
<li class="chapter" data-level="13" data-path="calculate-number-of-genes-per-megabase-per-chromosomes.html"><a href="calculate-number-of-genes-per-megabase-per-chromosomes.html"><i class="fa fa-check"></i><b>13</b> Calculate number of genes per megabase per chromosomes</a></li>
<li class="chapter" data-level="14" data-path="video-4.html"><a href="video-4.html"><i class="fa fa-check"></i><b>14</b> Video 4</a></li>
<li class="chapter" data-level="15" data-path="video-5.html"><a href="video-5.html"><i class="fa fa-check"></i><b>15</b> Video 5</a></li>
<li class="chapter" data-level="16" data-path="advanced-level.html"><a href="advanced-level.html"><i class="fa fa-check"></i><b>16</b> ADVANCED LEVEL</a></li>
<li class="chapter" data-level="17" data-path="video-6.html"><a href="video-6.html"><i class="fa fa-check"></i><b>17</b> Video 6</a></li>
<li class="chapter" data-level="18" data-path="video-7.html"><a href="video-7.html"><i class="fa fa-check"></i><b>18</b> Video 7</a></li>
<li class="chapter" data-level="19" data-path="ggplot.html"><a href="ggplot.html"><i class="fa fa-check"></i><b>19</b> GGPLOT</a>
<ul>
<li class="chapter" data-level="19.1" data-path="ggplot.html"><a href="ggplot.html#five-key-steps-to-ggplot"><i class="fa fa-check"></i><b>19.1</b> Five key steps to ggplot</a></li>
<li class="chapter" data-level="19.2" data-path="ggplot.html"><a href="ggplot.html#biology-behind-the-data-we-are-using"><i class="fa fa-check"></i><b>19.2</b> Biology behind the data we are using</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="analysis.html"><a href="analysis.html"><i class="fa fa-check"></i><b>20</b> Analysis</a>
<ul>
<li class="chapter" data-level="20.1" data-path="analysis.html"><a href="analysis.html#plotting"><i class="fa fa-check"></i><b>20.1</b> Plotting</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="analysis.html"><a href="analysis.html#base-set-up"><i class="fa fa-check"></i><b>20.1.1</b> Base set up</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="working-with-inbuilt-iris-data.html"><a href="working-with-inbuilt-iris-data.html"><i class="fa fa-check"></i><b>21</b> Working with inbuilt iris data</a></li>
<li class="chapter" data-level="22" data-path="r-markdown.html"><a href="r-markdown.html"><i class="fa fa-check"></i><b>22</b> R Markdown</a></li>
<li class="chapter" data-level="23" data-path="video-8.html"><a href="video-8.html"><i class="fa fa-check"></i><b>23</b> Video 8</a></li>
<li class="chapter" data-level="24" data-path="using-seqinr.html"><a href="using-seqinr.html"><i class="fa fa-check"></i><b>24</b> Using seqinr</a></li>
<li class="chapter" data-level="25" data-path="using-short-read.html"><a href="using-short-read.html"><i class="fa fa-check"></i><b>25</b> Using Short Read</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">YOUR TITLE HERE</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="video-8" class="section level1 hasAnchor" number="23">
<h1><span class="header-section-number">Chapter 23</span> Video 8<a href="video-8.html#video-8" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Check out this link to install <code>BiocManager</code> package
<a href="https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html" class="uri">https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html</a></p>
<p>Load the required libraries</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="video-8.html#cb269-1" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span></code></pre></div>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand</code></pre>
<pre><code>## The following objects are masked from &#39;package:data.table&#39;:
## 
##     first, second</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:data.table&#39;:
## 
##     shift</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<p>Read the sequence</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="video-8.html#cb290-1" tabindex="-1"></a>seq <span class="ot">=</span> <span class="fu">readDNAStringSet</span>(<span class="st">&quot;data/multifasta_seq.txt&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in .Call2(&quot;fasta_index&quot;, filexp_list, nrec, skip, seek.first.rec, :
## reading FASTA file data/multifasta_seq.txt: ignored 2545 invalid one-letter
## sequence codes</code></pre>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="video-8.html#cb292-1" tabindex="-1"></a>seq</span></code></pre></div>
<pre><code>## DNAStringSet object of length 6:
##     width seq                                               names               
## [1]   844 MVVVSSCVNTTRTAYTNSTRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV2_2019
## [2]   842 MTTSGSDDRCTTDDVANYTHTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV
## [3]   841 MVVVSSCVNTTRTAYTNSSTRGV...KGCCSCGSCCKDDDSVKGVKHYT Spike_Bat_coronav...
## [4]   834 MVVVSSCVNTTRTGGYTNSSTRG...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [5]   846 MHAVNSCVNTGRAASTNSSRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [6]   835 MKVVATVSSYTKCDDDRTANTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_WIV1_Bat_Co...</code></pre>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="video-8.html#cb294-1" tabindex="-1"></a>seq[[<span class="dv">1</span>]]</span></code></pre></div>
<pre><code>## 844-letter DNAString object
## seq: MVVVSSCVNTTRTAYTNSTRGVYYDKVRSSVHSTDS...MVTMCCMTSCCSCKGCCSCGSCCKDDDSVKGVKHYT</code></pre>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="video-8.html#cb296-1" tabindex="-1"></a>seq[[<span class="dv">2</span>]]</span></code></pre></div>
<pre><code>## 842-letter DNAString object
## seq: MTTSGSDDRCTTDDVANYTHTSSMRGVYYDRSDTYT...VMVTCCMTSCCSCKGACSCGSCCKDDDSVKGVKHYT</code></pre>
<p>Convert to character</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="video-8.html#cb298-1" tabindex="-1"></a><span class="fu">as.character</span>(seq[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>## [1] &quot;MVVVSSCVNTTRTAYTNSTRGVYYDKVRSSVHSTDSNVTWHAHVSGTNGTKRDNVNDGVYASTKSNRGWGTTDSKTSVNNATNVVKVCCNDGVYYHKNNKSWMSRVYSSANNCTYVSMDGKGNKNRVKNDGYKYSKHTNVRDGSAVDGNTRTAHRSYTGDSSSGWTAGAAAYYVGYRTKYNNGTTDAVDCADSTKCTKSTVKGYTSNRVTSVRNTNCGVNATRASVYAWNRKRSNCVADYSVYNSASSTKCYGVSTKNDCTNVYADSVRGDVRAGTGKADYNYKDDTGCVAWNSNNDSKVGGNYNYYRRKSNKRDSTYAGSTCNGVGNCYSYGTNGVGYYRVVVSHAATVCGKKSTNVKNKCVNNNGTGTGVTSNKKGRDADTTDAVRDTDTCSGGVSVTGTNTSNVAVYDVNCTVVAHADTTWRVYSTGSNVTRAGCGAHVNNSYCDGAGCASYTTNSRRARSVASSAYTMSGANSVAYSNNSATNTSVTTVSMTKTSVDCTMYCGDSTCSNYGSCTNRATGAVDKNTVAVKYKTKDGGNSDSKSKRSDNKVTADAGKYGDCGDAARDCAKNGTVTDMAYTSAAGTTSGWTGAGAAAMMAYRNGGVTNVYNKANNSAGKDSSSTASAGKDVVNNAANTVKSSNGASSVNDSRDKVAVDRTGRSTYVTRAARASANAATKMSCVGSKRVDCGKGYHMSSAHGVVHVTYVAKNTTAACHDGKAHRGVVSNGTHWVTRNYTTDNTVSGNCDVVGVNNTVYDDSKDKYKNHTSDVDGDSGNASVVNKDRNVAKNNSDGKYYKWWYWGAGAVMVTMCCMTSCCSCKGCCSCGSCCKDDDSVKGVKHYT&quot;</code></pre>
<p>Number of sequences</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="video-8.html#cb300-1" tabindex="-1"></a><span class="fu">length</span>(seq)</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<ul>
<li>Names of the sequences</li>
</ul>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="video-8.html#cb302-1" tabindex="-1"></a><span class="fu">names</span>(seq)</span></code></pre></div>
<pre><code>## [1] &quot;Spike_SARS_CoV2_2019&quot;          &quot;Spike_SARS_CoV&quot;               
## [3] &quot;Spike_Bat_coronavirus_RaTG13&quot;  &quot;Spike_Pangolin_coronavirus_GX&quot;
## [5] &quot;Spike_Pangolin_coronavirus_GD&quot; &quot;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<p>first 6 sequence</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="video-8.html#cb304-1" tabindex="-1"></a><span class="fu">head</span>(seq)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 6:
##     width seq                                               names               
## [1]   844 MVVVSSCVNTTRTAYTNSTRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV2_2019
## [2]   842 MTTSGSDDRCTTDDVANYTHTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV
## [3]   841 MVVVSSCVNTTRTAYTNSSTRGV...KGCCSCGSCCKDDDSVKGVKHYT Spike_Bat_coronav...
## [4]   834 MVVVSSCVNTTRTGGYTNSSTRG...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [5]   846 MHAVNSCVNTGRAASTNSSRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [6]   835 MKVVATVSSYTKCDDDRTANTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_WIV1_Bat_Co...</code></pre>
<ul>
<li>Reverse compliment of sequence</li>
</ul>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="video-8.html#cb306-1" tabindex="-1"></a>seq_rc <span class="ot">=</span> <span class="fu">reverseComplement</span>(seq[[<span class="dv">1</span>]])</span>
<span id="cb306-2"><a href="video-8.html#cb306-2" tabindex="-1"></a><span class="fu">head</span>(seq_rc)</span></code></pre></div>
<pre><code>## 6-letter DNAString object
## seq: ARDMBC</code></pre>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="video-8.html#cb308-1" tabindex="-1"></a><span class="fu">tail</span>(seq_rc)</span></code></pre></div>
<pre><code>## 6-letter DNAString object
## seq: SSBBBK</code></pre>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="video-8.html#cb310-1" tabindex="-1"></a><span class="fu">as.character</span>(seq_rc)</span></code></pre></div>
<pre><code>## [1] &quot;ARDMBCMBSHHHMGGSCGSGGCMGSGGSAKGGKABKBTCTCWRWWMRRMCHSNNMTBNYHMNBBSTNCSHCHBHSADNMRMHMSHHRBANNBCBBHGNCSBANHAARNYABWDACNSBBCYDTMCHDGTTAANMTBRABDBBCDTSSKDRCMCGHBYMSCBGSKMATTNTSTYTTYABRASYCAYHBTBMHYSHNBSSTCNSSMBANTTNNBBHMCTSTASSSHMCTSNNTMNRBNABCCNYRTKKTTTCTCAWCSAACTTSARTKHABACNMTGHYTTHCGHCRMCTHTABMNHSYMSMSHSNCCHMAMRMBTBANMHBTCATYNAGSCRNSGASHCGRKAGHBSAMAKSBAABSANATSNNSRTBSNTCSKARTSSTBSYTYYSNAARSTGCTCHGRSNNBDTCGCTYABNSCASRBYWAAHTDTBBAGNBHRBTBNSANACABSBCCSGAHAHYBTHAAHTHYCMMNSABCACACNNNBGMNMBNASMMCGBATTDSBBBYRRCBCNACRSRGNCBCNGASCTRASHYMNSMYYRRNRNCCBMSHNNSNWTBGCAHHMRNRHTMCACTYBHCYBSHTRBNAGHNMASBCRGMASSTSNRBSRHTBGNSYMYNWTRBSTYATNBCGNANYBSABYNSARCMBASMAGMASHTGHBTHAACNNRMAYRCBRRTTTCTAWCSSSHCARSYDTAYANCHBTSCHYBNADMSRMRCHNMBYNMNCMCHKSBRAGNNTSSRBYSKWSMNNMDRRBCHNGGBMBBNATNNBSAMSHAACWCYNSMASTRBCHNBNHYMACNACSBDTDWABNSHASDBSSYBMHRRBCYASNARTAYAANBGSSBBBK&quot;</code></pre>
<ul>
<li>Sequence composition -default - calculate the DNA composition</li>
</ul>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="video-8.html#cb312-1" tabindex="-1"></a>composition<span class="ot">=</span><span class="fu">alphabetFrequency</span>(seq)</span>
<span id="cb312-2"><a href="video-8.html#cb312-2" tabindex="-1"></a><span class="fu">print</span>(composition)</span></code></pre></div>
<pre><code>##       A  C  G   T  M  R  W   S  Y  K  V  H  D B  N - + .
## [1,] 79 40 82  97 14 42 12  99 54 61 97 17 62 0 88 0 0 0
## [2,] 84 39 78 100 20 39 11  96 54 60 91 15 74 0 81 0 0 0
## [3,] 80 40 82  98 14 39 12  97 55 63 94 18 64 0 85 0 0 0
## [4,] 76 40 85  98 15 41 12  94 55 61 97 17 61 0 82 0 0 0
## [5,] 82 40 81  99 15 40 12 100 60 58 95 14 60 0 90 0 0 0
## [6,] 82 39 77  93 17 42 11  98 50 60 93 15 72 0 86 0 0 0</code></pre>
<ul>
<li>Specific composition - lets say we wanna look for A, T, G, C and N</li>
</ul>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="video-8.html#cb314-1" tabindex="-1"></a>composition1<span class="ot">=</span><span class="fu">letterFrequency</span>(seq, <span class="at">letters=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;T&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;N&quot;</span>))</span>
<span id="cb314-2"><a href="video-8.html#cb314-2" tabindex="-1"></a><span class="fu">print</span>(composition1)</span></code></pre></div>
<pre><code>##       A   T  G  C  N
## [1,] 79  97 82 40 88
## [2,] 84 100 78 39 81
## [3,] 80  98 82 40 85
## [4,] 76  98 85 40 82
## [5,] 82  99 81 40 90
## [6,] 82  93 77 39 86</code></pre>
<p>About GC content - here we look for “G’ and”C”</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="video-8.html#cb316-1" tabindex="-1"></a>gc_content <span class="ot">&lt;-</span> <span class="fu">letterFrequency</span>(seq, <span class="at">letters=</span><span class="st">&quot;GC&quot;</span>, <span class="at">as.prob=</span><span class="cn">TRUE</span>)</span>
<span id="cb316-2"><a href="video-8.html#cb316-2" tabindex="-1"></a>gc_content</span></code></pre></div>
<pre><code>##            G|C
## [1,] 0.1445498
## [2,] 0.1389549
## [3,] 0.1450654
## [4,] 0.1498801
## [5,] 0.1430260
## [6,] 0.1389222</code></pre>
<p>Calculate the mean of Gc content</p>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="video-8.html#cb318-1" tabindex="-1"></a><span class="fu">mean</span>(gc_content)</span></code></pre></div>
<pre><code>## [1] 0.1433997</code></pre>
<p>Standard deviation</p>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="video-8.html#cb320-1" tabindex="-1"></a><span class="fu">sd</span>(gc_content)</span></code></pre></div>
<pre><code>## [1] 0.004148188</code></pre>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="video-8.html#cb322-1" tabindex="-1"></a><span class="fu">range</span>(gc_content)</span></code></pre></div>
<pre><code>## [1] 0.1389222 0.1498801</code></pre>
<p>Which GC has the maximum content</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="video-8.html#cb324-1" tabindex="-1"></a>seq[<span class="fu">which</span>(gc_content<span class="sc">==</span><span class="fu">max</span>(gc_content))]</span></code></pre></div>
<pre><code>## DNAStringSet object of length 1:
##     width seq                                               names               
## [1]   834 MVVVSSCVNTTRTGGYTNSSTRG...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...</code></pre>
<ul>
<li>Search for substrings</li>
</ul>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="video-8.html#cb326-1" tabindex="-1"></a>ECoR1<span class="ot">=</span><span class="fu">vcountPattern</span>(<span class="st">&quot;GAATTC&quot;</span>, seq)</span>
<span id="cb326-2"><a href="video-8.html#cb326-2" tabindex="-1"></a><span class="fu">print</span>(ECoR1)</span></code></pre></div>
<pre><code>## [1] 0 0 0 0 0 0</code></pre>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="video-8.html#cb328-1" tabindex="-1"></a>HindIII <span class="ot">=</span> <span class="fu">vcountPattern</span>(<span class="st">&quot;AAGTT&quot;</span>, seq)</span>
<span id="cb328-2"><a href="video-8.html#cb328-2" tabindex="-1"></a><span class="fu">print</span>(HindIII)</span></code></pre></div>
<pre><code>## [1] 1 0 1 1 1 0</code></pre>
<ul>
<li>Position of substrings</li>
</ul>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="video-8.html#cb330-1" tabindex="-1"></a>HindIII_pos<span class="ot">=</span><span class="fu">vmatchPattern</span>(<span class="st">&quot;AAGCT&quot;</span>, seq)</span>
<span id="cb330-2"><a href="video-8.html#cb330-2" tabindex="-1"></a><span class="fu">print</span>(HindIII_pos)</span></code></pre></div>
<pre><code>## MIndex object of length 6
## $Spike_SARS_CoV2_2019
## IRanges object with 0 ranges and 0 metadata columns:
##        start       end     width
##    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 
## $Spike_SARS_CoV
## IRanges object with 0 ranges and 0 metadata columns:
##        start       end     width
##    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 
## $Spike_Bat_coronavirus_RaTG13
## IRanges object with 0 ranges and 0 metadata columns:
##        start       end     width
##    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 
## ...
## &lt;3 more elements&gt;</code></pre>
<ul>
<li>Translate</li>
</ul>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="video-8.html#cb332-1" tabindex="-1"></a><span class="co"># translate(seq[[2]])</span></span>
<span id="cb332-2"><a href="video-8.html#cb332-2" tabindex="-1"></a><span class="co">#Check functions available</span></span>
<span id="cb332-3"><a href="video-8.html#cb332-3" tabindex="-1"></a><span class="fu">methods</span>(<span class="at">class =</span> <span class="st">&quot;DNAStringSet&quot;</span>)</span></code></pre></div>
<pre><code>##   [1] !                                     
##   [2] !=                                    
##   [3] [                                     
##   [4] [[                                    
##   [5] [[&lt;-                                  
##   [6] [&lt;-                                   
##   [7] %in%                                  
##   [8] &lt;                                     
##   [9] &lt;=                                    
##  [10] ==                                    
##  [11] &gt;                                     
##  [12] &gt;=                                    
##  [13] $                                     
##  [14] $&lt;-                                   
##  [15] aggregate                             
##  [16] alphabetFrequency                     
##  [17] anyDuplicated                         
##  [18] anyNA                                 
##  [19] append                                
##  [20] as.character                          
##  [21] as.complex                            
##  [22] as.data.frame                         
##  [23] as.env                                
##  [24] as.factor                             
##  [25] as.integer                            
##  [26] as.list                               
##  [27] as.logical                            
##  [28] as.matrix                             
##  [29] as.numeric                            
##  [30] as.raw                                
##  [31] as.vector                             
##  [32] bindROWS                              
##  [33] by                                    
##  [34] c                                     
##  [35] cbind                                 
##  [36] chartr                                
##  [37] coerce                                
##  [38] compact                               
##  [39] compareStrings                        
##  [40] complement                            
##  [41] consensusMatrix                       
##  [42] consensusString                       
##  [43] countOverlaps                         
##  [44] countPattern                          
##  [45] countPDict                            
##  [46] dinucleotideFrequencyTest             
##  [47] do.call                               
##  [48] droplevels                            
##  [49] duplicated                            
##  [50] elementMetadata                       
##  [51] elementMetadata&lt;-                     
##  [52] elementNROWS                          
##  [53] elementType                           
##  [54] eval                                  
##  [55] expand                                
##  [56] expand.grid                           
##  [57] extractAt                             
##  [58] extractROWS                           
##  [59] FactorToClass                         
##  [60] Filter                                
##  [61] findOverlaps                          
##  [62] getListElement                        
##  [63] hasOnlyBaseLetters                    
##  [64] head                                  
##  [65] ifelse2                               
##  [66] intersect                             
##  [67] is.na                                 
##  [68] is.unsorted                           
##  [69] isEmpty                               
##  [70] isMatchingEndingAt                    
##  [71] isMatchingStartingAt                  
##  [72] lapply                                
##  [73] length                                
##  [74] lengths                               
##  [75] letterFrequency                       
##  [76] make_XStringSet_from_strings          
##  [77] match                                 
##  [78] matchPattern                          
##  [79] matchPDict                            
##  [80] mcols                                 
##  [81] mcols&lt;-                               
##  [82] merge                                 
##  [83] mergeROWS                             
##  [84] metadata                              
##  [85] metadata&lt;-                            
##  [86] mstack                                
##  [87] names                                 
##  [88] names&lt;-                               
##  [89] nchar                                 
##  [90] neditEndingAt                         
##  [91] neditStartingAt                       
##  [92] normalizeSingleBracketReplacementValue
##  [93] nucleotideFrequencyAt                 
##  [94] oligonucleotideFrequency              
##  [95] order                                 
##  [96] overlapsAny                           
##  [97] PairwiseAlignments                    
##  [98] PairwiseAlignmentsSingleSubject       
##  [99] palindromeArmLength                   
## [100] parallel_slot_names                   
## [101] parallelVectorNames                   
## [102] pcompare                              
## [103] pcompareRecursively                   
## [104] PDict                                 
## [105] PWM                                   
## [106] rank                                  
## [107] rbind                                 
## [108] Reduce                                
## [109] relist                                
## [110] relistToClass                         
## [111] rename                                
## [112] rep                                   
## [113] rep.int                               
## [114] replaceAt                             
## [115] replaceLetterAt                       
## [116] replaceROWS                           
## [117] rev                                   
## [118] revElements                           
## [119] reverse                               
## [120] reverseComplement                     
## [121] sapply                                
## [122] selfmatch                             
## [123] seqinfo                               
## [124] seqinfo&lt;-                             
## [125] seqlevelsInUse                        
## [126] seqtype                               
## [127] seqtype&lt;-                             
## [128] setdiff                               
## [129] setequal                              
## [130] setListElement                        
## [131] shiftApply                            
## [132] show                                  
## [133] showAsCell                            
## [134] sort                                  
## [135] split                                 
## [136] split&lt;-                               
## [137] stack                                 
## [138] stringDist                            
## [139] strsplit                              
## [140] subseq                                
## [141] subseq&lt;-                              
## [142] subset                                
## [143] subsetByOverlaps                      
## [144] summary                               
## [145] table                                 
## [146] tail                                  
## [147] tapply                                
## [148] threebands                            
## [149] toString                              
## [150] transform                             
## [151] translate                             
## [152] trimLRPatterns                        
## [153] twoWayAlphabetFrequency               
## [154] union                                 
## [155] unique                                
## [156] uniqueLetters                         
## [157] unlist                                
## [158] unname                                
## [159] unsplit                               
## [160] unstrsplit                            
## [161] updateObject                          
## [162] values                                
## [163] values&lt;-                              
## [164] vcountPattern                         
## [165] vcountPDict                           
## [166] vmatchPattern                         
## [167] vwhichPDict                           
## [168] which.isMatchingEndingAt              
## [169] which.isMatchingStartingAt            
## [170] whichPDict                            
## [171] width                                 
## [172] window                                
## [173] window&lt;-                              
## [174] windows                               
## [175] with                                  
## [176] within                                
## [177] xtabs                                 
## [178] xtfrm                                 
## [179] xvcopy                                
## [180] zipdown                               
## see &#39;?methods&#39; for accessing help and source code</code></pre>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="r-markdown.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="using-seqinr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/balsaedi/RCourse/edit/main/08-video.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/balsaedi/RCourse/blob/main/08-video.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
